

<!DOCTYPE html>
<html lang="en" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/fluid.png">
  <link rel="icon" href="/img/fluid.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="Xuan Liu">
  <meta name="keywords" content="">
  
    <meta name="description" content="WGCNA文献和hdWGCNA文献阅读">
<meta property="og:type" content="article">
<meta property="og:title" content="WGCNA文献学习与配置">
<meta property="og:url" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/index.html">
<meta property="og:site_name" content="Hexo">
<meta property="og:description" content="WGCNA文献和hdWGCNA文献阅读">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/P1.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/P3.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/P4.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/P2.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/P5.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/P6.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/P7.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/P8.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/P9.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/ME_featureplot_scores.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/ME_correlogram.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/ME_dootploot.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/ME_vlnplot.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/INH-M12.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/hub_net.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/hubgene_umap.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/hubgene_umap_igraph.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/hub_umap.gif">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/test_DME_lollipop.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/test_DME_volcano2.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/ME_Trait_correlation_fdr.png">
<meta property="article:published_time" content="2023-05-22T12:54:15.000Z">
<meta property="article:modified_time" content="2024-05-24T09:07:37.382Z">
<meta property="article:author" content="Xuan Liu">
<meta property="article:tag" content="技术">
<meta property="article:tag" content="个人成长">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image" content="https://xuan-98-l.github.io/2023/05/22/WGCNA/P1.png">
  
  
  
  <title>WGCNA文献学习与配置 - Hexo</title>

  <link  rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css" />



  <link  rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css" />



<!-- 主题依赖的图标库，不要自行修改 -->
<!-- Do not modify the link that theme dependent icons -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/css/main.css" />


  <link id="highlight-css" rel="stylesheet" href="/css/highlight.css" />
  
    <link id="highlight-css-dark" rel="stylesheet" href="/css/highlight-dark.css" />
  




  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    Fluid.ctx = Object.assign({}, Fluid.ctx)
    var CONFIG = {"hostname":"xuan-98-l.github.io","root":"/","version":"1.9.4","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false,"scope":[]},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"left","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"code_language":{"enable":true,"default":"TEXT"},"copy_btn":true,"image_caption":{"enable":true},"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"placement":"right","headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":false,"follow_dnt":true,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":null,"app_key":null,"server_url":null,"path":"window.location.pathname","ignore_local":false}},"search_path":"/local-search.xml"};

    if (CONFIG.web_analytics.follow_dnt) {
      var dntVal = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
      Fluid.ctx.dnt = dntVal && (dntVal.startsWith('1') || dntVal.startsWith('yes') || dntVal.startsWith('on'));
    }
  </script>
  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
  


  
<meta name="generator" content="Hexo 6.3.0"></head>


<body>
  

  <header>
    

<div class="header-inner" style="height: 70vh;">
  <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/">
      <strong>Xuan</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/">
                <i class="iconfont icon-home-fill"></i>
                <span>Home</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/">
                <i class="iconfont icon-archive-fill"></i>
                <span>Archives</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/">
                <i class="iconfont icon-category-fill"></i>
                <span>Categories</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/">
                <i class="iconfont icon-tags-fill"></i>
                <span>Tags</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/">
                <i class="iconfont icon-user-fill"></i>
                <span>About</span>
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              <i class="iconfont icon-search"></i>
            </a>
          </li>
          
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">
              <i class="iconfont icon-dark" id="color-toggle-icon"></i>
            </a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

  

<div id="banner" class="banner" parallax=true
     style="background: url('/img/default.png') no-repeat center center; background-size: cover;">
  <div class="full-bg-img">
    <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
      <div class="banner-text text-center fade-in-up">
        <div class="h2">
          
            <span id="subtitle" data-typed-text="WGCNA文献学习与配置"></span>
          
        </div>

        
          
  <div class="mt-3">
    
    
      <span class="post-meta">
        <i class="iconfont icon-date-fill" aria-hidden="true"></i>
        <time datetime="2023-05-22 22:54" pubdate>
          May 22, 2023 pm
        </time>
      </span>
    
  </div>

  <div class="mt-1">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-chart"></i>
        
          20k words
        
      </span>
    

    
      <span class="post-meta mr-2">
        <i class="iconfont icon-clock-fill"></i>
        
        
        
          165 mins
        
      </span>
    

    
    
  </div>


        
      </div>

      
    </div>
  </div>
</div>

</div>

  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="side-col d-none d-lg-block col-lg-2">
      

    </div>

    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">WGCNA文献学习与配置</h1>
            
            
              <div class="markdown-body">
                
                <p>WGCNA文献和hdWGCNA文献阅读</p>
<span id="more"></span>
<h1 id="WGCNA介绍"><a href="#WGCNA介绍" class="headerlink" title="WGCNA介绍"></a>WGCNA介绍</h1><p><a target="_blank" rel="noopener" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559">WGCNA 文献</a></p>
<p><a target="_blank" rel="noopener" href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/">WGCNA 教程页面</a></p>
<p><a target="_blank" rel="noopener" href="https://peterlangfelder.com/articles/">作者 Peter Langfelder 博客</a></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/1804fb6d2ed8">一篇讲解清晰的blog</a></p>
<h2 id="研究背景"><a href="#研究背景" class="headerlink" title="研究背景"></a>研究背景</h2><p>Weighted correlation network analysis （WGCNA）是一种系统生物学方法，用于描述bulk RNA-seq中基因之间的相关性模式。WGCNA可用于寻找高度相关基因的聚类(module)，使用模块的特征基因或模块内hub基因对基因聚类情况进行汇总，将模块相互关联并与外部样本性状(使用特征基因网络方法)相关联，计算模块隶属度指标（模块与性状的关联）。该方法为基于网络的基因筛选方法提供了便利，可用于识别候选生物标志物或治疗靶点。这些方法已成功应用于多种生物背景，如癌症、小鼠遗传学、酵母遗传学、脑成像数据分析等。WGCNA R软件包是R函数的综合集合，包括网络构建、模块检测、基因选择、拓扑性质计算、数据模拟、可视化以及与外部软件接口等功能。</p>
<p>流程图<br><img src="/2023/05/22/WGCNA/P1.png" srcset="/img/loading.gif" lazyload></p>
<ol>
<li>构建基因共表达矩阵网络：计算基因间相关性<ul>
<li>基于Spearman method计算基因的相关性矩阵(<strong>Similarity Matrix</strong>)</li>
<li>将相关性矩阵转化为邻接矩阵(<strong>Adjacency Matrix</strong>)；Similarity Matrix被认为是undirected, weighted gene network，因此从Similarity Matrix转换到Adjacency Matrix的过程，被认为是从无向加权网络<strong>转移到scale-free network的过程</strong>。<ul>
<li>此步骤需要选择网络的类型(<strong>官方更推荐Signed或者Signed hybrid</strong>)<ul>
<li>Unsigned: Genes are similar if they are strongly correlated</li>
<li>Signed: Only applied to genes that are similar if they are <strong>positively correlated</strong></li>
<li>Signed hybrid: Negative associations are all treated as zero. (与signed network类似，与Signed network的差异可以很小)</li>
</ul>
</li>
<li>根据两个标准，选择<strong>soft thresholding β值</strong> (Power selected)；soft thresholding β值用于将Similarity Matrix转换到Adjacency Matrix<ul>
<li>Scale independence, 该图的点的纵坐标越高，越接近scale free network。</li>
<li>Mean Connectivity；由于在scale free network中，大多数节点的连通性较差，因此平均连通性越低，越接近scale free network。<br> <img src="/2023/05/22/WGCNA/P3.png" srcset="/img/loading.gif" lazyload></li>
</ul>
</li>
</ul>
</li>
<li>根据Adjacency Matrix绘制TOM plot (<strong>topological overlap matrix</strong>)<ul>
<li>基因网络的拓扑重叠热图(Heatmap plot of topological overlap in the gene network)是对基因相似性的评估。</li>
<li>在热图中，每一行和每一列对应一个基因，浅色表示低拓扑重叠，渐变深的红色表示高拓扑重叠。如果基因在网络中与相似的一组基因相连，那么它们被认为具有高度的拓扑重叠。</li>
</ul>
</li>
</ul>
</li>
<li>模块寻找：基于矩阵的数值进行成对比较，进行层次聚类，并通过Dynamic tree cut对聚类树进行模块划分<ul>
<li>模块的特征向量值signed module eigengene (ME)，反映了节点在模块内的隶属度。然后对特征基因高度相关的模块进行合并。<br><img src="/2023/05/22/WGCNA/P4.png" srcset="/img/loading.gif" lazyload></li>
</ul>
</li>
<li>将模块与表型相关联：寻找有生物学意义的模块<ul>
<li>具有高性状显著性的模块可能代表与样本性状相关的通路。</li>
</ul>
</li>
<li>模块间相关性研究（Optional）</li>
<li>寻找模块内部的key drivers（Optional）</li>
</ol>
<p>Tips:</p>
<p>Scale-free network 可轻易从无标度网络中寻找到hub genes（The highest-degree nodes）</p>
<p><img src="/2023/05/22/WGCNA/P2.png" srcset="/img/loading.gif" lazyload></p>
<p>官方建议删除Counts很低的基因（例如，删除所有在 90% 以上的样本中计数小于 10 的基因），因为这种低表达的Feature往往反映噪声,且基于计数的相关性大多为零并没有真正意义。实际阈值应基于实验设计、测序深度和样本计数。</p>
<p>无论是使用 RPKM、FPKM 还是简单的归一化计数，只要所有样本都以相同的方式处理，对 WGCNA 分析并没有太大影响。</p>
<p><a target="_blank" rel="noopener" href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/faq.html">WGCNA R包常见问题QA</a></p>
<h1 id="hdWGCNA"><a href="#hdWGCNA" class="headerlink" title="hdWGCNA"></a>hdWGCNA</h1><p><a target="_blank" rel="noopener" href="https://smorabit.github.io/hdWGCNA/">github</a><br><a target="_blank" rel="noopener" href="https://smorabit.github.io/hdWGCNA/articles/hdWGCNA.html">Tutorial</a></p>
<h2 id="1-初始化配置"><a href="#1-初始化配置" class="headerlink" title="1. 初始化配置"></a>1. 初始化配置</h2><p>注：在<em>SetupForWGCNA</em>步骤后，不建议对seurat对象进行subset（因为subset后WGCNA的输入基因可能无法代表原始的seurat对象）</p>
<p>gene_select参数的三种模式，选择不同的WGCNA输入基因：<br>variable: 使用Seurat对象中的高变基因<br>fraction: 使用在整个数据集或每组细胞中特定部分细胞中表达的基因，使用group.by指定。（在Demo中，我们将选择在这个数据集中至少5%的细胞中表达的基因。）<br>custom: 使用自定义列表中指定的基因。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs R">seurat_obj <span class="hljs-operator">&lt;-</span> SetupForWGCNA<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  gene_select <span class="hljs-operator">=</span> <span class="hljs-string">&quot;fraction&quot;</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># the gene selection approach</span><br>  fraction <span class="hljs-operator">=</span> <span class="hljs-number">0.05</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># fraction of cells that a gene needs to be expressed in order to be included</span><br>  wgcna_name <span class="hljs-operator">=</span> <span class="hljs-string">&quot;tutorial&quot;</span> <span class="hljs-comment"># the name of the hdWGCNA experiment</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h2 id="2-构建metacells"><a href="#2-构建metacells" class="headerlink" title="2. 构建metacells"></a>2. 构建metacells</h2><p>初始化配置后，需要从单细胞数据集构造构建<strong>metacells（元细胞）</strong>。</p>
<p>由于counts矩阵的稀疏性，会导致较大的方差，导致单细胞数据直接计算Pearson基因相关性会不准确。解决这个问题的一种方法是使用<a target="_blank" rel="noopener" href="https://github.com/tanaylab/metacells">metacells</a>的方法。</p>
<p>简单来说，<a target="_blank" rel="noopener" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1812-2">元细胞</a>是源自同一sampleid的小群相似细胞的聚集体。使用k近邻(KNN)算法找到 K 个最近邻居，识别相似细胞，并将它们进行聚合，然后计算这些细胞的平均或总表达，从而产生元细胞基因表达矩阵，使用元细胞的基因表达矩阵进行下游分析（分析思路类似细胞metadata矩阵，含有对细胞的分类信息等）。与Counts矩阵相比，元细胞表达矩阵的稀疏性大大降低，并提供了对转录状态更稳健的估计。因为像WGCNA等的相关网络方法对数据稀疏性很敏感，使用元细胞作为此类分析技术的输入会更稳健、噪音更小；且在处理大数据（例如分析数百万个单个细胞）时要分析的细胞数量减少（约 100 倍）。很多单细胞表观基因组方法，如Cicero，在构建共可及性网络之前采用了类似的元细胞聚集方法。</p>
<p>可参考：<a target="_blank" rel="noopener" href="https://divingintogeneticsandgenomics.com/post/how-to-do-gene-correlation-for-single-cell-rnaseq-data-part-2-using-meta-cell/">Tommy的blog</a></p>
<p>函数<em>MetacellsByGroups</em>，用于构建元细胞表达矩阵。该函数为元细胞数据集构建一个新的Seurat对象，该对象存储在hdWGCNA实验内部。该group.by参数确定将在哪些group中构造元细胞。我们只想从来自相同生物样本的细胞中构建元细胞，因此通过group.by将该信息传递给hdWGCNA至关重要。此外，我们通常为每种细胞类型分别构建元细胞。因此，在Demo中，我们按Sample和celltype进行分组，以获得期望的结果。</p>
<p>要聚合的单元格的数量k应该根据输入数据集的大小进行调整，通常较小的k可以用于小数据集。通常使用20到75之间的k值。Demo使用k&#x3D;25。元细胞之间允许的重叠细胞数量可以使用<em>max shared</em>参数进行调整。(这里的参数可以根据细胞数量调整k值)</p>
<p>注:当存在细胞数极少的细胞类型(由于极端细胞数导致代表性差的细胞类型)时，按照group.by&#x3D;celltype进行构建metacells的hdWGCNA表现不佳。例如，在Demo数据集中，脑血管细胞是细胞数最少的celltype，我们已经从这个分析中排除了它们。MetacellsByGroups有一个参数<em>min cells</em>，用于排除小于指定cell的group。如果最小cell的值过低，可能会出现错误。</p>
<p>hdWGCNA将group.by中所有输入的参数进行组合，形成新的metacell_grouping列，并要求每个metacell_grouping内的细胞数大于min_cells（默认值100），否则会被过滤掉。</p>
<p>K值的设置: <a target="_blank" rel="noopener" href="https://www.ibm.com/topics/knn#:~:text=The%20k%2Dnearest%20neighbors%20algorithm%2C%20also%20known%20as%20KNN%20or,of%20an%20individual%20data%20point.">kNN</a> k值越小，方差越大，偏差越小，k值越大，偏差越大，方差越小。k的选择很大程度上取决于输入数据，因为具有更多异常值或噪声的数据可能在k值较高时表现更好。建议<strong>k值设置为奇数</strong>。hdWGCNA推荐使用20到75之间的k值。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># construct metacells  in each group</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> MetacellsByGroups<span class="hljs-punctuation">(</span><br>  seurat_obj <span class="hljs-operator">=</span> seurat_obj<span class="hljs-punctuation">,</span><br>  group.by <span class="hljs-operator">=</span> <span class="hljs-built_in">c</span><span class="hljs-punctuation">(</span><span class="hljs-string">&quot;cell_type&quot;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&quot;Sample&quot;</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># specify the columns in seurat_obj@meta.data to group by</span><br>  reduction <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;harmony&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># select the dimensionality reduction to perform KNN on</span><br>  k <span class="hljs-operator">=</span> <span class="hljs-number">25</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># nearest-neighbors parameter</span><br>  max_shared <span class="hljs-operator">=</span> <span class="hljs-number">10</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># maximum number of shared cells between two metacells</span><br>  ident.group <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;cell_type&#x27;</span> <span class="hljs-comment"># set the Idents of the metacell seurat object</span><br><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># normalize metacell expression matrix:</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> NormalizeMetacells<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p>Tips: 参数<em>ident.group</em>会改变seurat_obj@misc$tutorial$<a href="mailto:&#119;&#103;&#x63;&#110;&#97;&#95;&#109;&#101;&#x74;&#97;&#99;&#x65;&#x6c;&#x6c;&#95;&#111;&#x62;&#x6a;&#x40;&#x61;&#x63;&#116;&#105;&#x76;&#x65;&#46;&#105;&#100;&#x65;&#110;&#x74;">&#119;&#103;&#x63;&#110;&#97;&#95;&#109;&#101;&#x74;&#97;&#99;&#x65;&#x6c;&#x6c;&#95;&#111;&#x62;&#x6a;&#x40;&#x61;&#x63;&#116;&#105;&#x76;&#x65;&#46;&#105;&#100;&#x65;&#110;&#x74;</a>的分组信息。seurat_obj@misc$tutorial$wgcna_metacell_obj是构建好metacells后的metadata信息，参数<em>group.by</em>会改变metadata的行名，若输入多个<em>group.by</em>参数，比如c(“cell_type”, “Sample”)会使得行名（细胞barcodes）变为”cell_type1#Sample1_1”,”cell_type1#Sample1_2”这种格式。</p>
<p><em>group.by</em>输入的参数必须包含<em>ident.group</em>的输入。</p>
<h2 id="3-共表达网络分析"><a href="#3-共表达网络分析" class="headerlink" title="3. 共表达网络分析"></a>3. 共表达网络分析</h2><h3 id="3-1-配置表达矩阵"><a href="#3-1-配置表达矩阵" class="headerlink" title="3.1 配置表达矩阵"></a>3.1 配置表达矩阵</h3><p>这里我们指定用于网络分析的表达式矩阵。因为我们只想包含抑制性神经元这群细胞，所以我们必须在构建网络之前对表达数据进行取子集。hdWGCNA包括<em>SetDatExpr</em>函数，用于存储将用于下游网络分析的一组给定细胞的转置表达矩阵。默认情况下使用元细胞表达矩阵(使用metacells&#x3D;TRUE)，但hdWGCNA确实允许在需要时使用单细胞表达矩阵。该函数允许用户指定从哪个data slot获取表达式矩阵，例如，如果用户希望应用SCTransform而不是NormalizeData的矩阵。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs R">seurat_obj <span class="hljs-operator">&lt;-</span> SetDatExpr<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  group_name <span class="hljs-operator">=</span> <span class="hljs-string">&quot;INH&quot;</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># the name of the group of interest in the group.by column</span><br>  group.by<span class="hljs-operator">=</span><span class="hljs-string">&#x27;cell_type&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># the metadata column containing the cell type info. This same column should have also been used in MetacellsByGroups</span><br>  assay <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;RNA&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># using RNA assay</span><br>  slot <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;data&#x27;</span> <span class="hljs-comment"># using normalized data</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p>同时对多个细胞类型或clusters执行共表达网络分析,需要向group_name参数传入字符向量。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs R">seurat_obj <span class="hljs-operator">&lt;-</span> SetDatExpr<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  group_name <span class="hljs-operator">=</span> <span class="hljs-built_in">c</span><span class="hljs-punctuation">(</span><span class="hljs-string">&quot;INH&quot;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&quot;EX&quot;</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">,</span><br>  group.by<span class="hljs-operator">=</span><span class="hljs-string">&#x27;cell_type&#x27;</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h3 id="3-2-soft-Power-软阈值-设置"><a href="#3-2-soft-Power-软阈值-设置" class="headerlink" title="3.2 soft-Power(软阈值)设置"></a>3.2 soft-Power(软阈值)设置</h3><p>WGCNA将相关性矩阵转化为邻接矩阵（给相关性进行幂次计算，使得基因间的连通性符合幂率分布规律）。选择合适的软阈值以减少相关矩阵中存在的噪声量，从而保留强连接并去除弱连接。</p>
<p>函数<em>TestSoftPowers</em>执行不同软阈值的参数扫描。该函数通过对不同power值的网络拓扑情况进行检查，帮助我们在构建共表达网络时指导选择软实力阈值。共同表达网络应该具有无标度的拓扑结构，因此TestSoftPowers函数对不同软阈值下共表达网络与无标度图的相似程度进行建模。此外，我们还包含了一个函数<em>PlotSoftPowers</em>来可视化参数扫描的结果。</p>
<p>此步骤的networkType的参数设置和图像输出与WGCNA相似。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># Test different soft powers:</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> TestSoftPowers<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  networkType <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;signed&#x27;</span> <span class="hljs-comment"># you can also use &quot;unsigned&quot; or &quot;signed hybrid&quot;</span><br><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># plot the results:</span><br>plot_list <span class="hljs-operator">&lt;-</span> PlotSoftPowers<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># assemble with patchwork</span><br>wrap_plots<span class="hljs-punctuation">(</span>plot_list<span class="hljs-punctuation">,</span> ncol<span class="hljs-operator">=</span><span class="hljs-number">2</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p><img src="/2023/05/22/WGCNA/P5.png" srcset="/img/loading.gif" lazyload></p>
<ul>
<li>Scale independence, 该图的点的纵坐标越高，越接近scale free network。<strong>纵坐标需要大于等于 0.8 才认为该网络是无标度拓扑模型。</strong></li>
<li>Mean Connectivity, Median Connectivity, Max Connectivity: 由于在scale free network中，大多数节点的连通性较差，因此这三个指标越低，越接近scale free network。</li>
</ul>
<p>如果用户没有提供软阈值，则<em>ConstructNetwork</em>将自动选择软阈值。软阈值参数扫描的输出表可以使用<em>GetPowerTable</em>函数查看</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs R">power_table <span class="hljs-operator">&lt;-</span> GetPowerTable<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><br>head<span class="hljs-punctuation">(</span>power_table<span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h3 id="3-3-构建共表达网络"><a href="#3-3-构建共表达网络" class="headerlink" title="3.3 构建共表达网络"></a>3.3 构建共表达网络</h3><p>使用hdWGCNA函数<em>ConstructNetwork</em>，它在底层调用WGCNA包的<em>blockwiseConsensusModules</em>函数。<em>blockwiseConsensusModules</em>的参数可以用相同的参数名直接传递给ConstructNetwork。下面的代码使用上面选择的软阈值来构建共表达网络。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># construct co-expression network:</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> ConstructNetwork<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span> soft_power<span class="hljs-operator">=</span><span class="hljs-number">9</span><span class="hljs-punctuation">,</span><br>  setDatExpr<span class="hljs-operator">=</span><span class="hljs-literal">FALSE</span><span class="hljs-punctuation">,</span><br>  tom_name <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;INH&#x27;</span> <span class="hljs-comment"># name of the topoligical overlap matrix written to disk</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p>函数<em>plotdendprogram</em>用于可视化WGCNA树状图，该图用于显示由网络分析产生的不同共表达模块。树状图上的每片叶子代表一个基因，底部的颜色表示共表达模块分配。<strong>与WGCNA相同，灰色模块由不属于任何共表达模块的基因组成。对于所有下游分析和解释，应该忽略灰色模块。</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs R">PlotDendrogram<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">,</span> main<span class="hljs-operator">=</span><span class="hljs-string">&#x27;INH hdWGCNA Dendrogram&#x27;</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p><img src="/2023/05/22/WGCNA/P6.png" srcset="/img/loading.gif" lazyload></p>
<p>3.4 topoligcal overlap matrix (TOM) 图<br>hdWGCNA将共表达网络表示为topoligcal overlap matrix (TOM)。这是一个基因对基因的方阵，其中每个值都是基因之间的拓扑重叠。当运行ConstructNetwork时，TOM被写入seurat对象，我们可以使用GetTOM函数将其加载到R中。高级用户可能希望检查TOM以进行自定义下游分析。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs R">TOM <span class="hljs-operator">&lt;-</span> GetTOM<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h2 id="4-模块特征基因与连通性"><a href="#4-模块特征基因与连通性" class="headerlink" title="4. 模块特征基因与连通性"></a>4. 模块特征基因与连通性</h2><h3 id="4-1-计算模块特征基因"><a href="#4-1-计算模块特征基因" class="headerlink" title="4.1 计算模块特征基因"></a>4.1 计算模块特征基因</h3><p>Module Eigengenes (MEs，模块特征基因)是一种用于度量整个共表达模块的基因表达谱的方式。简而言之，通过对包含每个模块的基因表达矩阵的子集执行(PCA)来计算模块特征基因。每个PCA矩阵的第一个PC是MEs。直观地说，它是一个“被合成的基因”，它是该模块中所有基因的最佳代表。</p>
<p>函数<em>ModuleEigengenes</em>用于计算单个细胞中的模块特征基因。此外，我们允许用户对MEs应用Harmony批量校正，从而产生harmonized module eigengenes(hMEs)。下面的代码使用group.by.vars参数执行由基于Sampleid进行Harmony的模块特征基因计算。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># need to run ScaleData first or else harmony throws an error:</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> ScaleData<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">,</span> features<span class="hljs-operator">=</span>VariableFeatures<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># compute all MEs in the full single-cell dataset</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> ModuleEigengenes<span class="hljs-punctuation">(</span><br> seurat_obj<span class="hljs-punctuation">,</span><br> group.by.vars<span class="hljs-operator">=</span><span class="hljs-string">&quot;Sample&quot;</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p>ME矩阵存储为矩阵，其中每行是一个细胞，每列是一个模块。可以使用<em>GetMEs</em>函数从Seurat对象中提取该矩阵，该函数在默认检索<strong>hMEs</strong>。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># harmonized module eigengenes:</span><br>hMEs <span class="hljs-operator">&lt;-</span> GetMEs<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># module eigengenes:</span><br>MEs <span class="hljs-operator">&lt;-</span> GetMEs<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">,</span> harmonized<span class="hljs-operator">=</span><span class="hljs-literal">FALSE</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h3 id="4-2-计算连通性"><a href="#4-2-计算连通性" class="headerlink" title="4.2 计算连通性"></a>4.2 计算连通性</h3><p>在共表达网络分析中，我们通常希望关注hub-gene，即在每个模块中连通性最高的基因。因此，我们希望<strong>确定每个基因的基于MEs的连通性，也称为kME</strong>。hdWGCNA使用<em>ModuleConnectivity</em>函数来计算完整的单细胞数据集中的kME值，而不是元细胞数据集。</p>
<p>这个函数本质上计算基因和模块特征基因之间的两两相关性。可以为数据集中的所有细胞计算kME，但我们建议在运行ConstructNetwork后的celltype或group中计算kME。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># compute eigengene-based connectivity (kME):</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> ModuleConnectivity<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  group.by <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;cell_type&#x27;</span><span class="hljs-punctuation">,</span> group_name <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;INH&#x27;</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p>可以根据模块具体的生物学意义对模块进行重命名。<br><a target="_blank" rel="noopener" href="https://smorabit.github.io/hdWGCNA/articles/customization.html">模块重命名说明书</a></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># rename the modules</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> ResetModuleNames<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  new_name <span class="hljs-operator">=</span> <span class="hljs-string">&quot;INH-M&quot;</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p>我们可以使用<em>PlotKMEs</em>函数可视化每个模块中按kME排序的基因。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># plot genes ranked by kME for each module</span><br>p <span class="hljs-operator">&lt;-</span> PlotKMEs<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">,</span> ncol<span class="hljs-operator">=</span><span class="hljs-number">5</span><span class="hljs-punctuation">)</span><br><br>p<br></code></pre></td></tr></table></figure>
<p><img src="/2023/05/22/WGCNA/P7.png" srcset="/img/loading.gif" lazyload></p>
<h3 id="4-3-获取模块分配表"><a href="#4-3-获取模块分配表" class="headerlink" title="4.3 获取模块分配表"></a>4.3 获取模块分配表</h3><p>hdWGCNA允许使用<em>GetModules</em>函数轻松访问模块分配表。</p>
<p>该表由三列组成:geneID，存储该基因的模块，模块颜色，这在许多下游绘图步骤中使用。如果调用<em>ModuleConnectivity</em>函数，则会为模块分配表添加额外的kME列。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># get the module assignment table:</span><br>modules <span class="hljs-operator">&lt;-</span> GetModules<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># show the first 6 columns:</span><br>head<span class="hljs-punctuation">(</span>modules<span class="hljs-punctuation">[</span><span class="hljs-punctuation">,</span><span class="hljs-number">1</span><span class="hljs-operator">:</span><span class="hljs-number">6</span><span class="hljs-punctuation">]</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>
<p><img src="/2023/05/22/WGCNA/P8.png" srcset="/img/loading.gif" lazyload></p>
<p>到此为止是hdWGCNA的主体分析流程，记得保存结果。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs R">saveRDS<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">,</span> file<span class="hljs-operator">=</span><span class="hljs-string">&#x27;hdWGCNA_object.rds&#x27;</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h3 id="4-4-计算hub-gene打分"><a href="#4-4-计算hub-gene打分" class="headerlink" title="4.4 计算hub-gene打分"></a>4.4 计算hub-gene打分</h3><p>Gene scoring是单细胞转录组学中一种常见的方法，用于计算一组基因的总体特征得分。Seurat使用<em>AddModuleScore</em>函数进行打分，但也有其他方法，如UCell。hdWGCNA使用<em>ModuleExprScore</em>函数，用于计算每个模块给定数量的基因的基因分数，基于Seurat或UCell算法。基因评分是通过计算模块特征基因来展示模块表达情况的另一种方法。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># compute gene scoring for the top 25 hub genes by kME for each module</span><br><span class="hljs-comment"># with Seurat method</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> ModuleExprScore<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  n_genes <span class="hljs-operator">=</span> <span class="hljs-number">25</span><span class="hljs-punctuation">,</span><br>  method<span class="hljs-operator">=</span><span class="hljs-string">&#x27;Seurat&#x27;</span><br><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># compute gene scoring for the top 25 hub genes by kME for each module</span><br><span class="hljs-comment"># with UCell method</span><br>library<span class="hljs-punctuation">(</span>UCell<span class="hljs-punctuation">)</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> ModuleExprScore<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  n_genes <span class="hljs-operator">=</span> <span class="hljs-number">25</span><span class="hljs-punctuation">,</span><br>  method<span class="hljs-operator">=</span><span class="hljs-string">&#x27;UCell&#x27;</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h2 id="5-使用Seurat进行基础的下游可视化"><a href="#5-使用Seurat进行基础的下游可视化" class="headerlink" title="5. 使用Seurat进行基础的下游可视化"></a>5. 使用Seurat进行基础的下游可视化</h2><h3 id="Featureplot"><a href="#Featureplot" class="headerlink" title="Featureplot"></a>Featureplot</h3><p><em>ModuleFeaturePlot</em>函数，用于为每个共表达模块构建由每个模块指定的唯一颜色着色的featureplot</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># make a featureplot of hMEs for each module</span><br>plot_list <span class="hljs-operator">&lt;-</span> ModuleFeaturePlot<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  features<span class="hljs-operator">=</span><span class="hljs-string">&#x27;hMEs&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># plot the hMEs</span><br>  order<span class="hljs-operator">=</span><span class="hljs-literal">TRUE</span> <span class="hljs-comment"># order so the points with highest hMEs are on top</span><br><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># stitch together with patchwork</span><br>wrap_plots<span class="hljs-punctuation">(</span>plot_list<span class="hljs-punctuation">,</span> ncol<span class="hljs-operator">=</span><span class="hljs-number">6</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>
<p><img src="/2023/05/22/WGCNA/P9.png" srcset="/img/loading.gif" lazyload></p>
<p>可以使用相同的函数绘制hub-gene的打分情况</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># make a featureplot of hub scores for each module</span><br>plot_list <span class="hljs-operator">&lt;-</span> ModuleFeaturePlot<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  features<span class="hljs-operator">=</span><span class="hljs-string">&#x27;scores&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># plot the hub gene scores</span><br>  order<span class="hljs-operator">=</span><span class="hljs-string">&#x27;shuffle&#x27;</span> <span class="hljs-comment"># order so cells are shuffled</span><br>  ucell <span class="hljs-operator">=</span> <span class="hljs-literal">TRUE</span> <span class="hljs-comment"># depending on Seurat vs UCell for gene scoring</span><br><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># stitch together with patchwork</span><br>wrap_plots<span class="hljs-punctuation">(</span>plot_list<span class="hljs-punctuation">,</span> ncol<span class="hljs-operator">=</span><span class="hljs-number">6</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>
<p><img src="/2023/05/22/WGCNA/ME_featureplot_scores.png" srcset="/img/loading.gif" lazyload></p>
<h3 id="模块相关性"><a href="#模块相关性" class="headerlink" title="模块相关性"></a>模块相关性</h3><p><em>ModuleCorrelogram</em>函数，可以使用R包corrplot根据每个模块的hMEs, MEs或hub基因评分来可视化每个模块之间的相关性。函数默认使用的是hMEs。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># plot module correlagram</span><br>ModuleCorrelogram<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p><img src="/2023/05/22/WGCNA/ME_correlogram.png" srcset="/img/loading.gif" lazyload></p>
<h3 id="DotPlot"><a href="#DotPlot" class="headerlink" title="DotPlot"></a>DotPlot</h3><p>将hMEs数据添加到metadata中进行绘制</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># get hMEs from seurat object</span><br>MEs <span class="hljs-operator">&lt;-</span> GetMEs<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">,</span> harmonized<span class="hljs-operator">=</span><span class="hljs-literal">TRUE</span><span class="hljs-punctuation">)</span><br>mods <span class="hljs-operator">&lt;-</span> colnames<span class="hljs-punctuation">(</span>MEs<span class="hljs-punctuation">)</span>; mods <span class="hljs-operator">&lt;-</span> mods<span class="hljs-punctuation">[</span>mods <span class="hljs-operator">!=</span> <span class="hljs-string">&#x27;grey&#x27;</span><span class="hljs-punctuation">]</span><br><br><span class="hljs-comment"># add hMEs to Seurat meta-data:</span><br>seurat_obj<span class="hljs-operator">@</span>meta.data <span class="hljs-operator">&lt;-</span> cbind<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-operator">@</span>meta.data<span class="hljs-punctuation">,</span> MEs<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment">#绘制dotplot</span><br><span class="hljs-comment"># plot with Seurat&#x27;s DotPlot function</span><br>p <span class="hljs-operator">&lt;-</span> DotPlot<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">,</span> features<span class="hljs-operator">=</span>mods<span class="hljs-punctuation">,</span> group.by <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;cell_type&#x27;</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># flip the x/y axes, rotate the axis labels, and change color scheme:</span><br>p <span class="hljs-operator">&lt;-</span> p <span class="hljs-operator">+</span><br>  coord_flip<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span><br>  RotatedAxis<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span><br>  scale_color_gradient2<span class="hljs-punctuation">(</span>high<span class="hljs-operator">=</span><span class="hljs-string">&#x27;red&#x27;</span><span class="hljs-punctuation">,</span> mid<span class="hljs-operator">=</span><span class="hljs-string">&#x27;grey95&#x27;</span><span class="hljs-punctuation">,</span> low<span class="hljs-operator">=</span><span class="hljs-string">&#x27;blue&#x27;</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># plot output</span><br>p<br></code></pre></td></tr></table></figure>
<p><img src="/2023/05/22/WGCNA/ME_dootploot.png" srcset="/img/loading.gif" lazyload></p>
<h3 id="Vlnplot"><a href="#Vlnplot" class="headerlink" title="Vlnplot"></a>Vlnplot</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># Plot INH-M4 hME using Seurat VlnPlot function</span><br>p <span class="hljs-operator">&lt;-</span> VlnPlot<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  features <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;INH-M12&#x27;</span><span class="hljs-punctuation">,</span><br>  group.by <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;cell_type&#x27;</span><span class="hljs-punctuation">,</span><br>  pt.size <span class="hljs-operator">=</span> <span class="hljs-number">0</span> <span class="hljs-comment"># don&#x27;t show actual data points</span><br><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># add box-and-whisker plots on top:</span><br>p <span class="hljs-operator">&lt;-</span> p <span class="hljs-operator">+</span> geom_boxplot<span class="hljs-punctuation">(</span>width<span class="hljs-operator">=</span><span class="hljs-number">.25</span><span class="hljs-punctuation">,</span> fill<span class="hljs-operator">=</span><span class="hljs-string">&#x27;white&#x27;</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># change axis labels and remove legend:</span><br>p <span class="hljs-operator">&lt;-</span> p <span class="hljs-operator">+</span> xlab<span class="hljs-punctuation">(</span><span class="hljs-string">&#x27;&#x27;</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span> ylab<span class="hljs-punctuation">(</span><span class="hljs-string">&#x27;hME&#x27;</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span> NoLegend<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># plot output</span><br>p<br></code></pre></td></tr></table></figure>
<p><img src="/2023/05/22/WGCNA/ME_vlnplot.png" srcset="/img/loading.gif" lazyload></p>
<h2 id="6-网络分析"><a href="#6-网络分析" class="headerlink" title="6. 网络分析"></a>6. 网络分析</h2><p><a target="_blank" rel="noopener" href="https://smorabit.github.io/hdWGCNA/articles/network_visualizations.html">链接</a></p>
<p>加载环境</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># single-cell analysis package</span><br>library<span class="hljs-punctuation">(</span>Seurat<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># plotting and data science packages</span><br>library<span class="hljs-punctuation">(</span>tidyverse<span class="hljs-punctuation">)</span><br>library<span class="hljs-punctuation">(</span>cowplot<span class="hljs-punctuation">)</span><br>library<span class="hljs-punctuation">(</span>patchwork<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># co-expression network analysis packages:</span><br>library<span class="hljs-punctuation">(</span>WGCNA<span class="hljs-punctuation">)</span><br>library<span class="hljs-punctuation">(</span>hdWGCNA<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># network analysis &amp; visualization package:</span><br>library<span class="hljs-punctuation">(</span>igraph<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># using the cowplot theme for ggplot</span><br>theme_set<span class="hljs-punctuation">(</span>theme_cowplot<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># set random seed for reproducibility</span><br>set.seed<span class="hljs-punctuation">(</span><span class="hljs-number">12345</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># load the Zhou et al snRNA-seq dataset</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> readRDS<span class="hljs-punctuation">(</span><span class="hljs-string">&#x27;data/Zhou_control.rds&#x27;</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h3 id="为每个模块生成单独的网络图"><a href="#为每个模块生成单独的网络图" class="headerlink" title="为每个模块生成单独的网络图"></a>为每个模块生成单独的网络图</h3><p><em>ModuleNetworkPlot</em>显示每个模块的单独网络图，按kME显示前25个基因。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs R">ModuleNetworkPlot<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>
<p><img src="/2023/05/22/WGCNA/INH-M12.png" srcset="/img/loading.gif" lazyload></p>
<p>在该网络中，每个节点代表一个基因，每条边代表网络中两个基因的共表达关系。每个模块网络图都是根据hdWGCNA模块分配表GetModules(seurat obj)中的颜色列进行着色的。kME排名前10位的枢纽基因位于图的中心，其余15个基因位于外圈。</p>
<p>可调节参数美化图片:</p>
<ul>
<li>edge.alpha：确定网络边缘的不透明度</li>
<li>vertex.size：确定节点的大小</li>
<li>vertex.label.cex：确定基因标签的字体尺寸</li>
</ul>
<h3 id="可视化由每个模块给定数量的hub-gens组成的网络"><a href="#可视化由每个模块给定数量的hub-gens组成的网络" class="headerlink" title="可视化由每个模块给定数量的hub-gens组成的网络"></a>可视化由每个模块给定数量的hub-gens组成的网络</h3><p><em>HubGeneNetworkPlot</em>可视化由每个模块给定数量的中心基因的所有模块组成的网络。该函数取用户指定的前n个枢纽基因和其他随机选择的基因，采用<a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Force-directed_graph_drawing">force-directed graph drawing algorithm 力导向布局算法</a>构建一个联合网络。</p>
<p>为了视觉清晰度，可以使用edge prop参数对网络中的边的数量进行下采样。在下面的例子中，我们将每个模块的前3个hub-gene和其他6个基因可视化。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># hubgene network</span><br>HubGeneNetworkPlot<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  n_hubs <span class="hljs-operator">=</span> <span class="hljs-number">3</span><span class="hljs-punctuation">,</span> n_other<span class="hljs-operator">=</span><span class="hljs-number">5</span><span class="hljs-punctuation">,</span><br>  edge_prop <span class="hljs-operator">=</span> <span class="hljs-number">0.75</span><span class="hljs-punctuation">,</span><br>  mods <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;all&#x27;</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p><img src="/2023/05/22/WGCNA/hub_net.png" srcset="/img/loading.gif" lazyload></p>
<p>每个节点代表一个基因，每个边代表一个共表达关系。在该网络中，我们用模块的颜色对模块内的边上色，对模块间的边上色。该网络中边缘的不透明度由共表达关系的强度来定义。附加的网络布局设置可以传递给图中<em>layout_with_fr</em>函数。用户还可以通过指定<em>return graph &#x3D; TRUE</em>来返回要绘图的igraph对象。</p>
<p>g &lt;- HubGeneNetworkPlot(seurat_obj,  return_graph&#x3D;TRUE)</p>
<h4 id="使用Umap同时可视化共表达中的所有基因"><a href="#使用Umap同时可视化共表达中的所有基因" class="headerlink" title="使用Umap同时可视化共表达中的所有基因"></a>使用Umap同时可视化共表达中的所有基因</h4><p><em>ModuleUMAPPlot</em>函数使用UMAP降维算法同时可视化共表达中的所有基因。</p>
<p>hdWGCNA包含<em>RunModuleUMAP</em>函数，基于TOM矩阵运行UMAP算法。对于UMAP分析，我们将TOM中的列子集设置为仅包含每个模块按kME排列的前n个中心基因，n由用户指定。因此，每个基因在UMAP空间中的位置取决于该基因与hub-gene的连通性。该函数利用了uwot R包中的UMAP实现，因此uwot:: UMAP函数的附加UMAP参数(如min dist或spread)可以在RunModuleUMAP中使用。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><code class="hljs R">seurat_obj <span class="hljs-operator">&lt;-</span> RunModuleUMAP<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  n_hubs <span class="hljs-operator">=</span> <span class="hljs-number">10</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># number of hub genes to include for the UMAP embedding</span><br>  n_neighbors<span class="hljs-operator">=</span><span class="hljs-number">15</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># neighbors parameter for UMAP</span><br>  min_dist<span class="hljs-operator">=</span><span class="hljs-number">0.1</span> <span class="hljs-comment"># min distance between points in UMAP space</span><br><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment">#可视化</span><br><span class="hljs-comment"># get the hub gene UMAP table from the seurat object</span><br>umap_df <span class="hljs-operator">&lt;-</span> GetModuleUMAP<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># plot with ggplot</span><br>ggplot<span class="hljs-punctuation">(</span>umap_df<span class="hljs-punctuation">,</span> aes<span class="hljs-punctuation">(</span>x<span class="hljs-operator">=</span>UMAP1<span class="hljs-punctuation">,</span> y<span class="hljs-operator">=</span>UMAP2<span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span><br>  geom_point<span class="hljs-punctuation">(</span><br>   color<span class="hljs-operator">=</span>umap_df<span class="hljs-operator">$</span>color<span class="hljs-punctuation">,</span> <span class="hljs-comment"># color each point by WGCNA module</span><br>   size<span class="hljs-operator">=</span>umap_df<span class="hljs-operator">$</span>kME<span class="hljs-operator">*</span><span class="hljs-number">2</span> <span class="hljs-comment"># size of each point based on intramodular connectivity</span><br>  <span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span><br>  umap_theme<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>
<p><img src="/2023/05/22/WGCNA/hubgene_umap.png" srcset="/img/loading.gif" lazyload></p>
<p>在这个图中，每个点代表一个基因。每个点的大小按其指定模块的基因的kME进行缩放。可以使用<strong>ModuleUMAPPlot</strong>函数来绘制基因及其共表达关系，并可视化模块UMAP中的基因（但并未可视化底层网络。）</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs R">ModuleUMAPPlot<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  edge.alpha<span class="hljs-operator">=</span><span class="hljs-number">0.25</span><span class="hljs-punctuation">,</span><br>  sample_edges<span class="hljs-operator">=</span><span class="hljs-literal">TRUE</span><span class="hljs-punctuation">,</span><br>  edge_prop<span class="hljs-operator">=</span><span class="hljs-number">0.1</span><span class="hljs-punctuation">,</span> <span class="hljs-comment"># proportion of edges to sample (20% here)</span><br>  label_hubs<span class="hljs-operator">=</span><span class="hljs-number">2</span> <span class="hljs-punctuation">,</span><span class="hljs-comment"># how many hub genes to plot per module?</span><br>  keep_grey_edges<span class="hljs-operator">=</span><span class="hljs-literal">FALSE</span><br><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>
<p><img src="/2023/05/22/WGCNA/hubgene_umap_igraph.png" srcset="/img/loading.gif" lazyload></p>
<p>这张图与我们用ggplot2绘制的图相似，但我们展示了共表达网络，并在每个模块中标记了2个hub-gene。为了视觉清晰度，我们使用edge prop参数下采样，只保留该网络中20%的边。我们还允许用户返回igraph对象来制作他们自己的自定义图或执行下游网络分析.</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#这个图可以优化，虽然只标记了两个hub gene，但仍然存在基因重叠现象</span><br>g <span class="hljs-operator">&lt;-</span> ModuleUMAPPlot<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">,</span>  return_graph<span class="hljs-operator">=</span><span class="hljs-literal">TRUE</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h3 id="可视化改变hub-gene的数量对uamp的影响"><a href="#可视化改变hub-gene的数量对uamp的影响" class="headerlink" title="可视化改变hub-gene的数量对uamp的影响"></a>可视化改变hub-gene的数量对uamp的影响</h3><p>我们在UMAP计算中包含的hub-gene的数量会影响下游可视化。在这里，我们使用gganimate来直观地比较用不同数量的hub-gene计算的umap。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># different label weights to test</span><br>n_hubs <span class="hljs-operator">&lt;-</span> <span class="hljs-built_in">c</span><span class="hljs-punctuation">(</span><span class="hljs-number">1</span><span class="hljs-punctuation">,</span> <span class="hljs-number">1</span><span class="hljs-operator">:</span><span class="hljs-number">10</span><span class="hljs-operator">*</span><span class="hljs-number">5</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># loop through different weights</span><br>df <span class="hljs-operator">&lt;-</span> data.frame<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span><br><span class="hljs-keyword">for</span><span class="hljs-punctuation">(</span>cur_hubs <span class="hljs-keyword">in</span> n_hubs<span class="hljs-punctuation">)</span><span class="hljs-punctuation">&#123;</span><br><br>  <span class="hljs-comment"># make a module UMAP using different label weights</span><br>  seurat_obj <span class="hljs-operator">&lt;-</span> RunModuleUMAP<span class="hljs-punctuation">(</span><br>    seurat_obj<span class="hljs-punctuation">,</span><br>    n_hubs <span class="hljs-operator">=</span> cur_hubs<span class="hljs-punctuation">,</span><br>    n_neighbors<span class="hljs-operator">=</span><span class="hljs-number">15</span><span class="hljs-punctuation">,</span><br>    exclude_grey <span class="hljs-operator">=</span> <span class="hljs-literal">TRUE</span><span class="hljs-punctuation">,</span><br>    min_dist<span class="hljs-operator">=</span><span class="hljs-number">0.1</span><br>  <span class="hljs-punctuation">)</span><br><br>  <span class="hljs-comment"># add to ongoing dataframe</span><br>  cur_df <span class="hljs-operator">&lt;-</span> GetModuleUMAP<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><br>  cur_df<span class="hljs-operator">$</span>n_hubs <span class="hljs-operator">&lt;-</span> cur_hubs<br>  df <span class="hljs-operator">&lt;-</span> rbind<span class="hljs-punctuation">(</span>df<span class="hljs-punctuation">,</span> cur_df<span class="hljs-punctuation">)</span><br><span class="hljs-punctuation">&#125;</span><br><br><span class="hljs-comment"># ggplot animation library</span><br>library<span class="hljs-punctuation">(</span>gganimate<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># plot with ggplot + gganimate</span><br>p <span class="hljs-operator">&lt;-</span> ggplot<span class="hljs-punctuation">(</span>df<span class="hljs-punctuation">,</span> aes<span class="hljs-punctuation">(</span>x<span class="hljs-operator">=</span>UMAP1<span class="hljs-punctuation">,</span> y<span class="hljs-operator">=</span>UMAP2<span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span><br>  geom_point<span class="hljs-punctuation">(</span>color<span class="hljs-operator">=</span>df<span class="hljs-operator">$</span>color<span class="hljs-punctuation">,</span> size<span class="hljs-operator">=</span>df<span class="hljs-operator">$</span>kME<span class="hljs-operator">*</span><span class="hljs-number">2</span> <span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span><br>  ggtitle<span class="hljs-punctuation">(</span><span class="hljs-string">&quot;N hubs: &#123;closest_state&#125;&quot;</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span><br>  transition_states<span class="hljs-punctuation">(</span><br>    n_hubs<span class="hljs-punctuation">,</span><br>    transition_length <span class="hljs-operator">=</span> <span class="hljs-number">2</span><span class="hljs-punctuation">,</span><br>    state_length <span class="hljs-operator">=</span> <span class="hljs-number">2</span><span class="hljs-punctuation">,</span><br>    wrap <span class="hljs-operator">=</span> <span class="hljs-literal">TRUE</span><br>  <span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span><br>  view_follow<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span><br>  enter_fade<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span><br>  umap_theme<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span><br><br>animate<span class="hljs-punctuation">(</span>p<span class="hljs-punctuation">,</span> fps<span class="hljs-operator">=</span><span class="hljs-number">30</span><span class="hljs-punctuation">,</span> duration<span class="hljs-operator">=</span><span class="hljs-number">25</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>
<p><img src="/2023/05/22/WGCNA/hub_umap.gif" srcset="/img/loading.gif" lazyload></p>
<h3 id="有监督的Umap-略"><a href="#有监督的Umap-略" class="headerlink" title="有监督的Umap,略"></a><a target="_blank" rel="noopener" href="https://smorabit.github.io/hdWGCNA/articles/network_visualizations.html">有监督的Umap,略</a></h3><h2 id="7-Differential-module-eigengene-DME-analysis"><a href="#7-Differential-module-eigengene-DME-analysis" class="headerlink" title="7. Differential module eigengene (DME) analysis"></a>7. Differential module eigengene (DME) analysis</h2><p>执行差异模块特征基因(DME)分析，揭示在给定group中上调或下调的模块。</p>
<p>首先加载环境</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># single-cell analysis package</span><br>library<span class="hljs-punctuation">(</span>Seurat<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># plotting and data science packages</span><br>library<span class="hljs-punctuation">(</span>tidyverse<span class="hljs-punctuation">)</span><br>library<span class="hljs-punctuation">(</span>cowplot<span class="hljs-punctuation">)</span><br>library<span class="hljs-punctuation">(</span>patchwork<span class="hljs-punctuation">)</span><br>library<span class="hljs-punctuation">(</span>ggrepel<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># co-expression network analysis packages:</span><br>library<span class="hljs-punctuation">(</span>WGCNA<span class="hljs-punctuation">)</span><br>library<span class="hljs-punctuation">(</span>hdWGCNA<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># using the cowplot theme for ggplot</span><br>theme_set<span class="hljs-punctuation">(</span>theme_cowplot<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># set random seed for reproducibility</span><br>set.seed<span class="hljs-punctuation">(</span><span class="hljs-number">12345</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># re-load the Zhou et al snRNA-seq dataset processed with hdWGCNA</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> readRDS<span class="hljs-punctuation">(</span><span class="hljs-string">&#x27;Zhou_2020_hdWGCNA.rds&#x27;</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h3 id="两组比较"><a href="#两组比较" class="headerlink" title="两组比较"></a>两组比较</h3><p>这里我们讨论如何在两个不同的组之间执行DME测试。使用hdWGCNA函数<em>FindDMEs</em>，这是Seurat函数FindMarkers的一个特例。使用Wilcoxon test，来比较两组，但其他test可以通过test.use参数调用。</p>
<p>由于教程数据集只包含对照大脑样本，我们将使用性别来定义我们的两个group。<em>FindDMEs</em>需要group1和group2的barcodes列表。此外，我们将只比较来自INH clusters的细胞，因为我们对INH clusters执行了网络分析。</p>
<h4 id="首先获取group1和group2的barcodes列表"><a href="#首先获取group1和group2的barcodes列表" class="headerlink" title="首先获取group1和group2的barcodes列表"></a>首先获取group1和group2的barcodes列表</h4><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs R">group1 <span class="hljs-operator">&lt;-</span> seurat_obj<span class="hljs-operator">@</span>meta.data <span class="hljs-operator">%&gt;%</span> subset<span class="hljs-punctuation">(</span>cell_type <span class="hljs-operator">==</span> <span class="hljs-string">&#x27;INH&#x27;</span> <span class="hljs-operator">&amp;</span> msex <span class="hljs-operator">==</span> <span class="hljs-number">0</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">%&gt;%</span> rownames<br>group2 <span class="hljs-operator">&lt;-</span> seurat_obj<span class="hljs-operator">@</span>meta.data <span class="hljs-operator">%&gt;%</span> subset<span class="hljs-punctuation">(</span>cell_type <span class="hljs-operator">==</span> <span class="hljs-string">&#x27;INH&#x27;</span> <span class="hljs-operator">&amp;</span> msex <span class="hljs-operator">!=</span> <span class="hljs-number">0</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">%&gt;%</span> rownames<br><br>head<span class="hljs-punctuation">(</span>group1<span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p>执行<em>FindDMEs</em></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><code class="hljs R">DMEs <span class="hljs-operator">&lt;-</span> FindDMEs<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  barcodes1 <span class="hljs-operator">=</span> group1<span class="hljs-punctuation">,</span><br>  barcodes2 <span class="hljs-operator">=</span> group2<span class="hljs-punctuation">,</span><br>  test.use<span class="hljs-operator">=</span><span class="hljs-string">&#x27;wilcox&#x27;</span><span class="hljs-punctuation">,</span><br>  wgcna_name<span class="hljs-operator">=</span><span class="hljs-string">&#x27;tutorial&#x27;</span><br><span class="hljs-punctuation">)</span><br><br>head<span class="hljs-punctuation">(</span>DMEs<span class="hljs-punctuation">)</span><br><br><br>p_val  avg_log2FC pct.1 pct.2    p_val_adj  module<br>INH<span class="hljs-operator">-</span>M18 <span class="hljs-number">4.714924e-23</span>  <span class="hljs-number">0.40637874</span> <span class="hljs-number">0.894</span> <span class="hljs-number">0.729</span> <span class="hljs-number">8.486863e-22</span> INH<span class="hljs-operator">-</span>M18<br>INH<span class="hljs-operator">-</span>M16 <span class="hljs-number">5.257311e-08</span> <span class="hljs-operator">-</span><span class="hljs-number">0.18257946</span> <span class="hljs-number">0.850</span> <span class="hljs-number">0.935</span> <span class="hljs-number">9.463160e-07</span> INH<span class="hljs-operator">-</span>M16<br>INH<span class="hljs-operator">-</span>M2  <span class="hljs-number">7.565615e-05</span> <span class="hljs-operator">-</span><span class="hljs-number">0.18746938</span> <span class="hljs-number">0.661</span> <span class="hljs-number">0.738</span> <span class="hljs-number">1.361811e-03</span>  INH<span class="hljs-operator">-</span>M2<br>INH<span class="hljs-operator">-</span>M15 <span class="hljs-number">1.496899e-03</span> <span class="hljs-operator">-</span><span class="hljs-number">0.06603535</span> <span class="hljs-number">0.969</span> <span class="hljs-number">0.977</span> <span class="hljs-number">2.694417e-02</span> INH<span class="hljs-operator">-</span>M15<br>INH<span class="hljs-operator">-</span>M10 <span class="hljs-number">1.513458e-02</span> <span class="hljs-operator">-</span><span class="hljs-number">0.07774661</span> <span class="hljs-number">0.975</span> <span class="hljs-number">0.980</span> <span class="hljs-number">2.724224e-01</span> INH<span class="hljs-operator">-</span>M10<br>INH<span class="hljs-operator">-</span>M17 <span class="hljs-number">2.034644e-02</span>  <span class="hljs-number">0.24035946</span> <span class="hljs-number">0.589</span> <span class="hljs-number">0.557</span> <span class="hljs-number">3.662360e-01</span> INH<span class="hljs-operator">-</span>M17<br></code></pre></td></tr></table></figure>

<h4 id="使用函数PlotDMEsLollipop对结果可视化"><a href="#使用函数PlotDMEsLollipop对结果可视化" class="headerlink" title="使用函数PlotDMEsLollipop对结果可视化"></a>使用函数<em>PlotDMEsLollipop</em>对结果可视化</h4><p><a target="_blank" rel="noopener" href="https://cran.r-project.org/web/packages/forestploter/vignettes/forestploter-intro.html">森林图绘制</a></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#这个图可以优化一下，有点丑</span><br>PlotDMEsLollipop<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span> <br>  DMEs<span class="hljs-punctuation">,</span> <br>  wgcna_name<span class="hljs-operator">=</span><span class="hljs-string">&#x27;tutorial&#x27;</span><span class="hljs-punctuation">,</span> <br>  pvalue <span class="hljs-operator">=</span> <span class="hljs-string">&quot;p_val_adj&quot;</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p><img src="/2023/05/22/WGCNA/test_DME_lollipop.png" srcset="/img/loading.gif" lazyload></p>
<p>该图显示了每个模块的fold change，每个点的大小对应于该模块中基因的数量, X表示该点没有达到统计显著性。</p>
<p>对于PlotDMEsLollipop，如果我们在DMEs的data frame中有其他自定义列，我们可以使用group.by为一个比较组或用于绘图的比较组列表提供列名和比较参数。例如:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><code class="hljs R">PlotDMEslollipop<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span> <br>  DMEs<span class="hljs-punctuation">,</span> <br>  wgcna_name<span class="hljs-operator">=</span><span class="hljs-string">&#x27;tutorial&#x27;</span><span class="hljs-punctuation">,</span> <br>  group.by <span class="hljs-operator">=</span> <span class="hljs-string">&quot;Comparisons&quot;</span><span class="hljs-punctuation">,</span> <br>  comparison <span class="hljs-operator">=</span> <span class="hljs-built_in">c</span><span class="hljs-punctuation">(</span><span class="hljs-string">&quot;group1_vs_control&quot;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&quot;group2_vs_control&quot;</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">,</span>  <br>  pvalue <span class="hljs-operator">=</span> <span class="hljs-string">&quot;p_val_adj&quot;</span><br><span class="hljs-punctuation">)</span> <br></code></pre></td></tr></table></figure>



<h4 id="使用函数PlotDMEsVolcano将结果可视化。"><a href="#使用函数PlotDMEsVolcano将结果可视化。" class="headerlink" title="使用函数PlotDMEsVolcano将结果可视化。"></a>使用函数<em>PlotDMEsVolcano</em>将结果可视化。</h4><p>使用<strong>PlotDMEsVolcano</strong>来制作火山图，以显示fold change和显著性水平。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs R">PlotDMEsVolcano<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  DMEs<span class="hljs-punctuation">,</span><br>  wgcna_name <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;tutorial&#x27;</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h3 id="一对多的DME分析"><a href="#一对多的DME分析" class="headerlink" title="一对多的DME分析"></a>一对多的DME分析</h3><p>与Seurat函数FindAllMarkers类似，在指定要分组cell的列时，可以使用函数<strong>FindAllDMEs</strong>执行单对全的DME测试。我们在这里分组。通过每个细胞类型进行一对一的测试。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs R">group.by <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;cell_type&#x27;</span><br><br>DMEs_all <span class="hljs-operator">&lt;-</span> FindAllDMEs<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  group.by <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;cell_type&#x27;</span><span class="hljs-punctuation">,</span><br>  wgcna_name <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;tutorial&#x27;</span><br><span class="hljs-punctuation">)</span><br><br>head<span class="hljs-punctuation">(</span>DMEs_all<span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p>绘图</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><code class="hljs R">p <span class="hljs-operator">&lt;-</span> PlotDMEsVolcano<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  DMEs_all<span class="hljs-punctuation">,</span><br>  wgcna_name <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;tutorial&#x27;</span><span class="hljs-punctuation">,</span><br>  plot_labels<span class="hljs-operator">=</span><span class="hljs-literal">FALSE</span><span class="hljs-punctuation">,</span><br>  show_cutoff<span class="hljs-operator">=</span><span class="hljs-literal">FALSE</span><br><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># facet wrap by each cell type</span><br>p <span class="hljs-operator">+</span> facet_wrap<span class="hljs-punctuation">(</span><span class="hljs-operator">~</span>group<span class="hljs-punctuation">,</span> ncol<span class="hljs-operator">=</span><span class="hljs-number">3</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>
<p><img src="/2023/05/22/WGCNA/test_DME_volcano2.png" srcset="/img/loading.gif" lazyload></p>
<h2 id="8-模块特征相关性分析"><a href="#8-模块特征相关性分析" class="headerlink" title="8.模块特征相关性分析"></a>8.模块特征相关性分析</h2><h3 id="计算相关性"><a href="#计算相关性" class="headerlink" title="计算相关性"></a>计算相关性</h3><p>使用函数<em>ModuleTraitCorrelation</em>将选定的变量与模块特征基因相关联。该函数计算指定分组的相关性，因为我们可以预期某些变量可能与分组中的某些模块相关，但与其他单元组中的模块无关。在这里要注意输入变量的数据类型：</p>
<p>可以使用的变量包括：</p>
<ul>
<li>数值变量</li>
<li>只有 2 个类别的分类变量，例如“CONTROL”和“CASE”。</li>
<li>具有levels的分类变量，数据格式需要为因子型数据。例如，您可能有一个“疾病阶段”类别，按“健康”、“阶段 1”、“阶段 2”、“阶段 3”等排序。</li>
</ul>
<p>不能使用的变量：</p>
<p>具有两个以上无排序的分类变量。比如“sampleid”变量，由于这个变量的排序不一定有生物学意义，因此这样的变量不适合模块性状相关分析。再举个例子：假设有一个由三种转基因小鼠品系和一种对照小鼠组成的数据集。由于没法对这个分类进行排序，在运行相关性之前，必须将分类变量转换为数字，因此最终会得到完全没有生物学意义的相关性。在这种情况下，应该分别在对照和每个转基因小鼠品系之间建立成对的相关性比较。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># convert sex to factor </span><br>seurat_obj<span class="hljs-operator">$</span>msex <span class="hljs-operator">&lt;-</span> as.factor<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-operator">$</span>msex<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># convert age_death to numeric</span><br>seurat_obj<span class="hljs-operator">$</span>age_death <span class="hljs-operator">&lt;-</span> <span class="hljs-built_in">as.numeric</span><span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-operator">$</span>age_death<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment"># list of traits to correlate，检测模块与&#x27;braaksc&#x27;, &#x27;pmi&#x27;, &#x27;msex&#x27;, &#x27;age_death&#x27;, &#x27;doublet_scores&#x27;, &#x27;nCount_RNA&#x27;, &#x27;nFeature_RNA&#x27;, &#x27;total_counts_mt&#x27;的相关性</span><br>cur_traits <span class="hljs-operator">&lt;-</span> <span class="hljs-built_in">c</span><span class="hljs-punctuation">(</span><span class="hljs-string">&#x27;braaksc&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&#x27;pmi&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&#x27;msex&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&#x27;age_death&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&#x27;doublet_scores&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&#x27;nCount_RNA&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&#x27;nFeature_RNA&#x27;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&#x27;total_counts_mt&#x27;</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment">#按照cell_type对图片进行分面</span><br>seurat_obj <span class="hljs-operator">&lt;-</span> ModuleTraitCorrelation<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  traits <span class="hljs-operator">=</span> cur_traits<span class="hljs-punctuation">,</span><br>  group.by<span class="hljs-operator">=</span><span class="hljs-string">&#x27;cell_type&#x27;</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p>对于使用的任何分类变量，使用<em>ModuleTraitCorrelation</em>函数都会打印出一条警告消息，可忽视该警告，警告只是为了确保输入的分类变量有生物学意义。</p>
<h3 id="检查输出"><a href="#检查输出" class="headerlink" title="检查输出"></a>检查输出</h3><p>mt_cor是一个列表；cor保存相关结果，pval保存相关 p 值，fdr保存 FDR 校正的 p 值。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment"># get the mt-correlation results</span><br>mt_cor <span class="hljs-operator">&lt;-</span> GetModuleTraitCorrelation<span class="hljs-punctuation">(</span>seurat_obj<span class="hljs-punctuation">)</span><br><br><span class="hljs-built_in">names</span><span class="hljs-punctuation">(</span>mt_cor<span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<h3 id="绘制相关热图"><a href="#绘制相关热图" class="headerlink" title="绘制相关热图"></a>绘制相关热图</h3><p>使用<em>PlotModuleTraitCorrelation</em>函数绘制相关性分析的结果。该函数为每个相关性矩阵创建一个单独的热图，然后进行拼图。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><code class="hljs R">PlotModuleTraitCorrelation<span class="hljs-punctuation">(</span><br>  seurat_obj<span class="hljs-punctuation">,</span><br>  label <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;fdr&#x27;</span><span class="hljs-punctuation">,</span><br>  label_symbol <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;stars&#x27;</span><span class="hljs-punctuation">,</span><br>  text_size <span class="hljs-operator">=</span> <span class="hljs-number">2</span><span class="hljs-punctuation">,</span><br>  text_digits <span class="hljs-operator">=</span> <span class="hljs-number">2</span><span class="hljs-punctuation">,</span><br>  text_color <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;white&#x27;</span><span class="hljs-punctuation">,</span><br>  high_color <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;yellow&#x27;</span><span class="hljs-punctuation">,</span><br>  mid_color <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;black&#x27;</span><span class="hljs-punctuation">,</span><br>  low_color <span class="hljs-operator">=</span> <span class="hljs-string">&#x27;purple&#x27;</span><span class="hljs-punctuation">,</span><br>  plot_max <span class="hljs-operator">=</span> <span class="hljs-number">0.2</span><span class="hljs-punctuation">,</span><br>  combine<span class="hljs-operator">=</span><span class="hljs-literal">TRUE</span><br><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>

<p><img src="/2023/05/22/WGCNA/ME_Trait_correlation_fdr.png" srcset="/img/loading.gif" lazyload></p>
<h1 id="hdWGCNA与WGCNA-R包比较"><a href="#hdWGCNA与WGCNA-R包比较" class="headerlink" title="hdWGCNA与WGCNA R包比较"></a>hdWGCNA与WGCNA R包比较</h1><ol>
<li>使用WGCNA分析scRNA数据时，需要提取表达量矩阵进行分析。hdWGCNA可直接对接Seurat R包，直接基于Seurat结果进行下游分析：每个clusters随机抽取20%的细胞，可以选择使用全部的基因或者仅使用seurat挑选出来的高变基因，以及不同的降维方式和维度数，用于共表达模块的WGCNA检测，选取软阈值。中间部分与WGCNA无区别。最后，在单细胞空间中计算不同模块的平均表达式，并绘制降维图。生成所有模块的图表示，并可选择进行对应的GO富集分析。</li>
<li>基于metacells重新进行normalize后的数据进行分析,考虑到单细胞矩阵稀疏的特点，增加了数据的稳健性。</li>
<li>下游可视化形式更加丰富，除了WGCNA经典的模块与性状相关性热图，模块识别聚类树图等，增加了结合Seurat进行可视化，网络分析，差异模块特征基因（类似差异分析）分析，富集分析（图片有点丑），将生成的模块信息投影到其他数据集（可以将单细胞数据生成的模块信息投影到ATAC-seq,空转或者其他物种的单细胞数据集中）的内容。</li>
<li>hdWGCNA中的TOM图可视化需要手动实现。</li>
<li>hdWGCNA主要集中在对模块特征基因的展示，并以umap形式提供模块中全部基因的展示（与WGCNA生成的聚类图箱线图和柱状图相比，在细胞数和基因数较多时可视化方式更友好）。</li>
</ol>
<h1 id="hdWGCNA耗时统计"><a href="#hdWGCNA耗时统计" class="headerlink" title="hdWGCNA耗时统计"></a>hdWGCNA耗时统计</h1><p>demo数据 36601基因，36671细胞</p>
<ul>
<li>SetUpForWGCN初始化设置耗时：1.690009 mins</li>
<li>Meta细胞构建: 6.070287 mins</li>
<li>SetDatExpr运行时间: 0.6600406 secs</li>
<li>软阈值选择: 50.69805 secs</li>
<li>构建共表达网络耗时：5.24069 mins</li>
<li><strong>模块特征基因</strong>： 1.20085 hours</li>
</ul>
<h1 id="复习关键名词"><a href="#复习关键名词" class="headerlink" title="复习关键名词"></a>复习关键名词</h1><ul>
<li>ME：通过对包含每个模块的基因表达矩阵执行(PCA)来计算获得的“伪基因”。每个PCA矩阵的第一个PC是MEs。</li>
<li>kME: 是根据MEs的连通性的排名，排名越高，就证明这个基因是共表达网络中高度连接的成员，主要用于筛选hub-gene.</li>
<li>hME: 对MEs使用Harmony对sampleid矫正后获得的值。在基础可视化部分默认用到。</li>
</ul>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/37937a27c77e">参考链接</a></p>

                
              </div>
            
            <hr/>
            <div>
              <div class="post-metas my-3">
  
    <div class="post-meta mr-3 d-flex align-items-center">
      <i class="iconfont icon-category"></i>
      

<span class="category-chains">
  
  
    
      <span class="category-chain">
        
  <a href="/categories/%E8%BD%AF%E4%BB%B6%E7%AE%97%E6%B3%95/" class="category-chain-item">软件算法</a>
  
  
    <span>></span>
    
  <a href="/categories/%E8%BD%AF%E4%BB%B6%E7%AE%97%E6%B3%95/%E7%94%9F%E4%BF%A1/" class="category-chain-item">生信</a>
  
  

  

      </span>
    
  
</span>

    </div>
  
  
    <div class="post-meta">
      <i class="iconfont icon-tags"></i>
      
        <a href="/tags/%E6%8A%80%E6%9C%AF/">#技术</a>
      
        <a href="/tags/%E4%B8%AA%E4%BA%BA%E6%88%90%E9%95%BF/">#个人成长</a>
      
    </div>
  
</div>


              
  

  <div class="license-box my-3">
    <div class="license-title">
      <div>WGCNA文献学习与配置</div>
      <div>https://xuan-98-l.github.io/2023/05/22/WGCNA/</div>
    </div>
    <div class="license-meta">
      
        <div class="license-meta-item">
          <div>Author</div>
          <div>Xuan Liu</div>
        </div>
      
      
        <div class="license-meta-item license-meta-date">
          <div>Posted on</div>
          <div>May 22, 2023</div>
        </div>
      
      
      
        <div class="license-meta-item">
          <div>Licensed under</div>
          <div>
            
              
              
                <a target="_blank" href="https://creativecommons.org/licenses/by/4.0/">
                  <span class="hint--top hint--rounded" aria-label="BY - Attribution">
                    <i class="iconfont icon-by"></i>
                  </span>
                </a>
              
            
          </div>
        </div>
      
    </div>
    <div class="license-icon iconfont"></div>
  </div>



              
                <div class="post-prevnext my-3">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/2023/06/05/Zotero/" title="Zotero配置">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">Zotero配置</span>
                        <span class="visible-mobile">Previous</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2023/05/22/SCENIC/" title="基于SCENIC的转录因子分析原理讲解与实操">
                        <span class="hidden-mobile">基于SCENIC的转录因子分析原理讲解与实操</span>
                        <span class="visible-mobile">Next</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>

    <div class="side-col d-none d-lg-block col-lg-2">
      
  <aside class="sidebar" style="margin-left: -1rem">
    <div id="toc">
  <p class="toc-header">
    <i class="iconfont icon-list"></i>
    <span>Table of Contents</span>
  </p>
  <div class="toc-body" id="toc-body"></div>
</div>



  </aside>


    </div>
  </div>
</div>





  



  



  



  



  







    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">Search</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v" for="local-search-input">Keyword</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>

    

    
  </main>

  <footer>
    <div class="footer-inner">
  
    <div class="footer-content">
       <a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a> <i class="iconfont icon-love"></i> <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener"><span>Fluid</span></a> 
    </div>
  
  
  
  
</div>

  </footer>

  <!-- Scripts -->
  
  <script  src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://lib.baomitu.com/jquery/3.6.0/jquery.min.js" ></script>
<script  src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js" ></script>
<script  src="/js/events.js" ></script>
<script  src="/js/plugins.js" ></script>


  <script  src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var subtitle = document.getElementById('subtitle');
      if (!subtitle || !typing) {
        return;
      }
      var text = subtitle.getAttribute('data-typed-text');
      
        typing(text);
      
    })(window, document);
  </script>




  
    <script  src="/js/img-lazyload.js" ></script>
  




  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/tocbot/4.18.2/tocbot.min.js', function() {
    var toc = jQuery('#toc');
    if (toc.length === 0 || !window.tocbot) { return; }
    var boardCtn = jQuery('#board-ctn');
    var boardTop = boardCtn.offset().top;

    window.tocbot.init(Object.assign({
      tocSelector     : '#toc-body',
      contentSelector : '.markdown-body',
      linkClass       : 'tocbot-link',
      activeLinkClass : 'tocbot-active-link',
      listClass       : 'tocbot-list',
      isCollapsedClass: 'tocbot-is-collapsed',
      collapsibleClass: 'tocbot-is-collapsible',
      scrollSmooth    : true,
      includeTitleTags: true,
      headingsOffset  : -boardTop,
    }, CONFIG.toc));
    if (toc.find('.toc-list-item').length > 0) {
      toc.css('visibility', 'visible');
    }

    Fluid.events.registerRefreshCallback(function() {
      if ('tocbot' in window) {
        tocbot.refresh();
        var toc = jQuery('#toc');
        if (toc.length === 0 || !tocbot) {
          return;
        }
        if (toc.find('.toc-list-item').length > 0) {
          toc.css('visibility', 'visible');
        }
      }
    });
  });
</script>


  <script src=https://lib.baomitu.com/clipboard.js/2.0.11/clipboard.min.js></script>

  <script>Fluid.plugins.codeWidget();</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js', function() {
    window.anchors.options = {
      placement: CONFIG.anchorjs.placement,
      visible  : CONFIG.anchorjs.visible
    };
    if (CONFIG.anchorjs.icon) {
      window.anchors.options.icon = CONFIG.anchorjs.icon;
    }
    var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
    var res = [];
    for (var item of el) {
      res.push('.markdown-body > ' + item.trim());
    }
    if (CONFIG.anchorjs.placement === 'left') {
      window.anchors.options.class = 'anchorjs-link-left';
    }
    window.anchors.add(res.join(', '));

    Fluid.events.registerRefreshCallback(function() {
      if ('anchors' in window) {
        anchors.removeAll();
        var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
        var res = [];
        for (var item of el) {
          res.push('.markdown-body > ' + item.trim());
        }
        if (CONFIG.anchorjs.placement === 'left') {
          anchors.options.class = 'anchorjs-link-left';
        }
        anchors.add(res.join(', '));
      }
    });
  });
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js', function() {
    Fluid.plugins.fancyBox();
  });
</script>


  <script>Fluid.plugins.imageCaption();</script>

  <script  src="/js/local-search.js" ></script>





<!-- 主题的启动项，将它保持在最底部 -->
<!-- the boot of the theme, keep it at the bottom -->
<script  src="/js/boot.js" ></script>


  

  <noscript>
    <div class="noscript-warning">Blog works best with JavaScript enabled</div>
  </noscript>
</body>
</html>
