

<!DOCTYPE html>
<html lang="en" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/fluid.png">
  <link rel="icon" href="/img/fluid.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="Xuan Liu">
  <meta name="keywords" content="">
  
    <meta name="description" content="Integrated transcriptome landscape of ALS identifies genome instability linked to TDP-43 pathology">
<meta property="og:type" content="article">
<meta property="og:title" content="Integrated transcriptome landscape of ALS identifies genome instability linked to TDP-43 pathology">
<meta property="og:url" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/index.html">
<meta property="og:site_name" content="Hexo">
<meta property="og:description" content="Integrated transcriptome landscape of ALS identifies genome instability linked to TDP-43 pathology">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020183848022.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020204825146.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020210001416.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020210925550.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020215604044.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020222512985.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020224325364.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020224530785.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020230608339.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020231721860.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020231936356.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020232052439.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020232304821.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020234208517.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020234908792.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020235249322.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020235048783.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020235504303.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020235343196.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241021000919713.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241021001700232.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241021001802404.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241021003134057.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241021003032637.png">
<meta property="article:published_time" content="2024-10-20T13:34:43.000Z">
<meta property="article:modified_time" content="2024-10-20T13:36:57.278Z">
<meta property="article:author" content="Xuan Liu">
<meta property="article:tag" content="技术">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image" content="https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020183848022.png">
  
  
  
  <title>Integrated transcriptome landscape of ALS identifies genome instability linked to TDP-43 pathology - Hexo</title>

  <link  rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css" />



  <link  rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css" />



<!-- 主题依赖的图标库，不要自行修改 -->
<!-- Do not modify the link that theme dependent icons -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/css/main.css" />


  <link id="highlight-css" rel="stylesheet" href="/css/highlight.css" />
  
    <link id="highlight-css-dark" rel="stylesheet" href="/css/highlight-dark.css" />
  




  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    Fluid.ctx = Object.assign({}, Fluid.ctx)
    var CONFIG = {"hostname":"xuan-98-l.github.io","root":"/","version":"1.9.4","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false,"scope":[]},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"left","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"code_language":{"enable":true,"default":"TEXT"},"copy_btn":true,"image_caption":{"enable":true},"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"placement":"right","headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":false,"follow_dnt":true,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":null,"app_key":null,"server_url":null,"path":"window.location.pathname","ignore_local":false}},"search_path":"/local-search.xml"};

    if (CONFIG.web_analytics.follow_dnt) {
      var dntVal = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
      Fluid.ctx.dnt = dntVal && (dntVal.startsWith('1') || dntVal.startsWith('yes') || dntVal.startsWith('on'));
    }
  </script>
  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
  


  
<meta name="generator" content="Hexo 6.3.0"></head>


<body>
  

  <header>
    

<div class="header-inner" style="height: 70vh;">
  <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/">
      <strong>Xuan</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/">
                <i class="iconfont icon-home-fill"></i>
                <span>Home</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/">
                <i class="iconfont icon-archive-fill"></i>
                <span>Archives</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/">
                <i class="iconfont icon-category-fill"></i>
                <span>Categories</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/">
                <i class="iconfont icon-tags-fill"></i>
                <span>Tags</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/">
                <i class="iconfont icon-user-fill"></i>
                <span>About</span>
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              <i class="iconfont icon-search"></i>
            </a>
          </li>
          
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">
              <i class="iconfont icon-dark" id="color-toggle-icon"></i>
            </a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

  

<div id="banner" class="banner" parallax=true
     style="background: url('/img/default.png') no-repeat center center; background-size: cover;">
  <div class="full-bg-img">
    <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
      <div class="banner-text text-center fade-in-up">
        <div class="h2">
          
            <span id="subtitle" data-typed-text="Integrated transcriptome landscape of ALS identifies genome instability linked to TDP-43 pathology"></span>
          
        </div>

        
          
  <div class="mt-3">
    
    
      <span class="post-meta">
        <i class="iconfont icon-date-fill" aria-hidden="true"></i>
        <time datetime="2024-10-21 00:34" pubdate>
          October 21, 2024 am
        </time>
      </span>
    
  </div>

  <div class="mt-1">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-chart"></i>
        
          12k words
        
      </span>
    

    
      <span class="post-meta mr-2">
        <i class="iconfont icon-clock-fill"></i>
        
        
        
          103 mins
        
      </span>
    

    
    
  </div>


        
      </div>

      
    </div>
  </div>
</div>

</div>

  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="side-col d-none d-lg-block col-lg-2">
      

    </div>

    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">Integrated transcriptome landscape of ALS identifies genome instability linked to TDP-43 pathology</h1>
            
            
              <div class="markdown-body">
                
                <p>Integrated transcriptome landscape of ALS identifies genome instability linked to TDP-43 pathology</p>
<span id="more"></span>
<h1 id="Why-read-this-paper"><a href="#Why-read-this-paper" class="headerlink" title="Why read this paper?"></a><strong>Why read this paper?</strong></h1><p>Focuse on ALS research using the same key technologies: human induced pluripotent stem cell-derived motor neurons (iPSMNs) and RNA sequencing.</p>
<h1 id="Which-journal"><a href="#Which-journal" class="headerlink" title="Which journal?"></a><strong>Which journal?</strong></h1><p>Nature communication (Published: 20 April 2023)</p>
<h1 id="Who-wrote-the-paper"><a href="#Who-wrote-the-paper" class="headerlink" title="Who wrote the paper?"></a><strong>Who wrote the paper?</strong></h1><p><strong>Dr Oliver J. Ziff</strong>: The Francis Crick Institute, London. Research focus on investigating the causes of Amyotrophic Lateral Sclerosis (ALS) using bioinformatic multi-omic approaches.</p>
<p><strong>Professor Professor Rickie Patan</strong>i: University College London, London. A physician scientist with over a decade of direct experience working on human induced pluripotent stem cell （iPSMNs) models of neurodegeneration.</p>
<h1 id="What-is-ALS"><a href="#What-is-ALS" class="headerlink" title="What is ALS"></a>What is ALS</h1><ul>
<li>Amyotrophic lateral sclerosis (ALS), is a nervous system disease that affects nerve cells in the brain and spinal cord. ALS causes loss of muscle control. ALS often begins with muscle twitching and weakness in an arm or leg, trouble swallowing or slurred speech. Eventually ALS affects chewing, swallowing, speaking and breathing. The disease gets worse over time.<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020183848022.png" srcset="/img/loading.gif" lazyload alt="Fig 1 (Ref 1)"></li>
<li>ALS cases can be grouped by two categories:</li>
<li>About 10% of ALS cases are inherited and referred to as <strong>familial ALS</strong>. More than 20 gene mutations have been established to cause ALS, the most common being in C9orf72, SOD1, TARDBP and FUS.</li>
<li><strong>Sporadic ALS</strong> technically refers to cases without a family history, the term is sometimes also used to describe cases without a identifiable genetic basis.</li>
</ul>
<h1 id="Background"><a href="#Background" class="headerlink" title="Background"></a>Background</h1><ul>
<li><strong>Post-mortem tissue</strong> has been used to study ALS. TDP-43 proteinopathy remains a post-mortem pathologic diagnosis of ALS,  which is observed in 97% of cases and is characterised by the mislocalisation and aggregation of TDP-43 in the cytoplasm of neurons.</li>
<li>A major hurdle in identifying the causes of ALS is the inaccessibility of patient motor neuron. Post-mortem tissue represents the end stages of the disease and contains very few surviving motor neurons. </li>
<li>The ability to differentiate patient-derived induced pluripotent stem cells (iPSCs) into motor neurons (iPSMNs) has transformed ALS modeling. <strong>Human induced pluripotent stem cell-derived motor neurons (iPSMNs)</strong> allow researchers to model ALS motor neuron degeneration at earlier stages and enable sporadic ALS to be modelled.</li>
</ul>
<h1 id="Study-Objective"><a href="#Study-Objective" class="headerlink" title="Study Objective"></a>Study Objective</h1><p>This research uses data from ALS iPSMNs and post-mortem tissue to perform an integrated transcriptome analysis. The goal is to understand how various genetic backgrounds contribute to ALS and to map the landscape of RNA changes related to the disease. </p>
<h1 id="Cohort-Design"><a href="#Cohort-Design" class="headerlink" title="Cohort Design"></a>Cohort Design</h1><p><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020204825146.png" srcset="/img/loading.gif" lazyload></p>
<h2 id="iPSMN-cohort"><a href="#iPSMN-cohort" class="headerlink" title="iPSMN cohort:"></a>iPSMN cohort:</h2><p>Online database search (from GEO,SRA, EBI arrayExpress, ENA, synapse.org, etc.) identified 16 ALS iPSMN bulk RNA-sequencing (RNA-seq) datasets, which undergone short-read bulk RNA-sequencing (RNA-seq) from human iPSMN samples derived from individuals with ALS and non-ALS controls.</p>
<p>After quality control, 15 dataset remained:<br>In total 429 iPSMNs, of which 323 were from ALS patients and 106 were from non-ALS controls.</p>
<p>Quality control (Excluded datasets that:)<br>(1) had not undergone an accepted spinal motor neuron differentiation <strong>protocol</strong> using the steps detailed in Sances et al.<br>(2) failed <strong>RNA-seq quality control</strong> measures<br>(3) inadequate <strong>neuronal marker expression</strong><br>(4) exhibited <strong>unadjustable batch effects</strong> between ALS and control samples (e.g. different RNA library strategies or sequencing platforms, such as ALS (Ribo-Zero) and control (polyA) samples).</p>
<p>Samples clustered into two groups based on library preparation (poly-adenylated or total ribosomal depletion).<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020210001416.png" srcset="/img/loading.gif" lazyload alt="Supplementary Figure 2"><br>Scatterplot of PC1 against PC2 individual gene loadings of the top 500 most variable genes: Principal component gene loadings confirmed that the separation was driven by <strong>histone</strong> and <strong>small nucleolar encoding genes</strong>, which represent non-polyadenylated genes.<br>(Loading plot can be used to identify which variables have the strongest influence on each component. The coefficients can range from -1 to 1. Coefficients close to -1 or 1 indicate that the variable has a strong influence on the component.)<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020210925550.png" srcset="/img/loading.gif" lazyload></p>
<p>ALS iPSMNs carried pathogenic mutations in 10 different genes, including C9orf72 (n &#x3D; 60), SOD1 (n &#x3D; 20), FUS(n &#x3D; 14), TARDBP (n &#x3D; 10), VCP (n &#x3D; 3) and other ALS mutations (n &#x3D; 9)<br>Sporadic ALS(patients without an identifiable ALS mutation): 208 (64.2%)<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020215604044.png" srcset="/img/loading.gif" lazyload></p>
<h2 id="Postmotrem-cohort"><a href="#Postmotrem-cohort" class="headerlink" title="Postmotrem cohort"></a>Postmotrem cohort</h2><p><strong>Post-mortem spinal cord ALS RNA-seq samples</strong> were derived from samples from the New York Genome Centre (NYGC) ALS consortium, consisting of tissue from 214 ALS patients and 57 controls.<br>Samples from non-spinal cord sites were excluded.</p>
<p>Post-mortem subgroup analyses were limited by the sample size for <strong>FUS (n &#x3D; 2)</strong> and there were <strong>no TARDBP mutants</strong> available.</p>
<h1 id="Results"><a href="#Results" class="headerlink" title="Results"></a>Results</h1><h2 id="ALS-iPSMNs-activate-the-DNA-damage-response-Pan-ALS-gene-expression-changes"><a href="#ALS-iPSMNs-activate-the-DNA-damage-response-Pan-ALS-gene-expression-changes" class="headerlink" title="ALS iPSMNs activate the DNA damage response(Pan-ALS gene expression changes)"></a>ALS iPSMNs activate the DNA damage response(<strong>Pan-ALS</strong> gene expression changes)</h2><h3 id="All-library-preparation-together"><a href="#All-library-preparation-together" class="headerlink" title="All library preparation together:"></a>All library preparation together:</h3><h5 id="Differential-expression-gene-DEG"><a href="#Differential-expression-gene-DEG" class="headerlink" title="Differential expression gene (DEG)"></a>Differential expression gene (DEG)</h5><p>DESeq2<br>323 ALS vs 106 control iPSMNs<br>The Wald test, controlling for sex differences and dataset variation with the <strong>design formula ∼ sex + dataset + condition</strong>.</p>
<p><strong>Upregulated</strong>:<br>20</p>
<p><strong>Downregulated</strong>:<br>23</p>
<p>Amongst differentially expressed genes most increased in ALS was the endoribonuclease RNase L (<strong>RNASEL</strong>) which regulates the decay of cytoplasmic RNA and localisation of RNA binding proteins (RBPs)<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020222512985.png" srcset="/img/loading.gif" lazyload></p>
<h5 id="Functional-over-representation-analysis-ORA"><a href="#Functional-over-representation-analysis-ORA" class="headerlink" title="Functional over-representation analysis (ORA)"></a>Functional over-representation analysis (ORA)</h5><p><strong>Upregulated</strong>:<br>the DNA damage response (hypergeometric FDR &#x3D; 2.2 × 10−5; SESN1, RRM2B, TNFRSF10B) p53 signalling (FDR &#x3D; 2.7 × 10−5; CDKN1A, TP53TG3E, FBXO22)</p>
<p><strong>Downregulated</strong>:<br>DNA-binding transcription factor activity (FDR &#x3D; 0.003; MYOG, TBX5, POU5F1)<br>ventral spinal cord development (FDR &#x3D; 0.004; LMO4, OLIG2, FOXN4; Fig. 2b).<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020224325364.png" srcset="/img/loading.gif" lazyload></p>
<h5 id="Gene-Set-Enrichment-Analysis-GSEA"><a href="#Gene-Set-Enrichment-Analysis-GSEA" class="headerlink" title="Gene Set Enrichment Analysis (GSEA)"></a>Gene Set Enrichment Analysis (GSEA)</h5><p><strong>Upregulated</strong>:<br>p53 signal transduction gene set (GO:0072331, n &#x3D; 264)<br>The mitotic G1 DNA damage checkpoint (daughter pathway of DNA damage response)<br>Intrinsic apoptosis signalling (daughter pathway of intrinsic apoptotic signaling pathway by p53 class mediator).</p>
<h5 id="Signalling-Pathway-RespOnsive-GENes-PROGENy-analysis"><a href="#Signalling-Pathway-RespOnsive-GENes-PROGENy-analysis" class="headerlink" title="Signalling Pathway RespOnsive GENes (PROGENy) analysis"></a>Signalling Pathway RespOnsive GENes (PROGENy) analysis</h5><p>PROGENy is resource that leverages a large compendium of publicly available signaling perturbation experiments to yield a common core of pathway responsive genes. For each pathway, a collection of genes are available along their contribution and significance to it.</p>
<p>Inside PROGENy, one can find gene signatures for 14 different pathways: Androgen，Estrogen，EGFR，Hypoxia，JAK-STAT，<strong>MAPK</strong>，NFkB，PI3K，<strong>p53</strong>，TGFb，TNFa，Trail，VEGF and <strong>WNT</strong></p>
<p><strong>Upregulated</strong>:<br>p53 (NES + 13.0, p &lt; 0.001): Examining each gene in the p53 pathway according to its p53 weighting in PROGENy revealed that the genes with the strongest responsiveness in p53 activity in ALS iPSMNs included CDKN1A, SESN1, RRM2B, MDM2, C2orf66, ZNF561 and ZMAT3 (Fig. 2e).<br>Mitogen-Activated Protein Kinase (MAPK; NES + 5.6, p &lt; 0.001)</p>
<p><strong>Downregulated</strong>:<br>WNT (NES −2.5, p &#x3D; 0.03; Fig. 2d)</p>
<p><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020224530785.png" srcset="/img/loading.gif" lazyload></p>
<h5 id="Transcription-factors-activity-within-the-DoRothEA-database"><a href="#Transcription-factors-activity-within-the-DoRothEA-database" class="headerlink" title="Transcription factors activity (within the DoRothEA database)"></a>Transcription factors activity (within the DoRothEA database)</h5><p>The activities of 429 transcription factors (TFs) from their regulon expression within the DoRothEA database.<br>Upregulated:<br>TP53 was the TF with the greatest increase in activity in ALS (NES + 7.62, p &lt; 0.001)<br>ZNF274<br>ATF4</p>
<p>Downregulated:<br>PRDM14<br>ZNF263<br>SIX5<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020230608339.png" srcset="/img/loading.gif" lazyload></p>
<h3 id="subgroup-analysis-in-library-preparation-separately"><a href="#subgroup-analysis-in-library-preparation-separately" class="headerlink" title="subgroup analysis in library preparation separately"></a>subgroup analysis in library preparation separately</h3><p>subgroup analysis in poly(A) datasets and total Ribo-Zero separately:</p>
<p><strong>Poly(A) datasets: (10 datasets; 48 ALS, 43 control iPSMNs)</strong><br><strong>Total Ribo-Zero (5 datasets; 275 ALS, 63 control iPSMNs)</strong></p>
<p><strong>Differential expression gene (DEG):</strong> poly(A) datasets there were 69, and in total Ribo-Zero datasets there were 12<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020231721860.png" srcset="/img/loading.gif" lazyload><br><strong>ORA:</strong> RNase L was significantly increased in ALS iPSMNs in both library preparation analyses independently.<br><strong>PROGENy</strong>: significant increases in <strong>p53</strong> pathway activity in ALS with both library preparations<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020231936356.png" srcset="/img/loading.gif" lazyload><br>Activity of ranscription factors within the DoRothEA database:<br><strong>TP53</strong> TF activity was significantly increased in ALS iPSMNs in both library preparation groups<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020232052439.png" srcset="/img/loading.gif" lazyload></p>
<h3 id="subgroup-analysis-in-dataset-separately"><a href="#subgroup-analysis-in-dataset-separately" class="headerlink" title="subgroup analysis in dataset separately"></a>subgroup analysis in dataset separately</h3><p>Despite this, both TP53 transcription factor and p53 signalling were independently upregulated in ALS iPSMNs in 11 of 17 datasets (seems a mistake? why 17 datasets).<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020232304821.png" srcset="/img/loading.gif" lazyload></p>
<h3 id="Proteome-in-Answer-ALS-dataset"><a href="#Proteome-in-Answer-ALS-dataset" class="headerlink" title="Proteome in Answer ALS dataset"></a>Proteome in Answer ALS dataset</h3><h5 id="DEP"><a href="#DEP" class="headerlink" title="DEP:"></a><strong>DEP</strong>:</h5><p>no proteins were significant at FDR &lt; 0.05 in ALS compared to control<br>276 were significantly different at unadjusted p &lt; 0.05 (including p53 pathway components RBBP7, CSNK2B, PRMT1, CNOT9)</p>
<h5 id="ORA"><a href="#ORA" class="headerlink" title="ORA:"></a><strong>ORA</strong>:</h5><p><strong>Upregulated</strong>:<br>protein metabolism (e.g. PSDMD9, NAE1, PSMB5)<br>RNA metabolism (e.g. APP, CSTF1, CIRBP, SNRPD2)<br>protein binding (e.g. RRBP1, RPS29, EIF1)<br>other processes established in ALS pathophysiology (stress response, cholesterol synthesis, nucleocytoplasmic transport) </p>
<p>Downregulated:<br>Golgi transport</p>
<h5 id="GSEA"><a href="#GSEA" class="headerlink" title="GSEA"></a>GSEA</h5><p>p53 pathway nonsignificant increase (NES + 0.98, p &#x3D; 0.5)</p>
<h5 id="Correlation-between-mRNA-and-protein-level"><a href="#Correlation-between-mRNA-and-protein-level" class="headerlink" title="Correlation between mRNA and protein level"></a>Correlation between mRNA and protein level</h5><p>Comparing changes in mRNA expression with protein expression revealed a weak inverse correlation, consistent with previous reports showing poor correlations between mRNA and protein<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020234208517.png" srcset="/img/loading.gif" lazyload></p>
<h2 id="p53-activation-across-different-ALS-genetic-backgrounds"><a href="#p53-activation-across-different-ALS-genetic-backgrounds" class="headerlink" title="p53 activation across different ALS genetic backgrounds"></a>p53 activation across <strong>different ALS genetic backgrounds</strong></h2><p>Of the 15 datasets, 7 included <strong>C9orf72</strong> mutants (comprising 60 C9orf72 iPSMNs and 83 control iPSMNs), 5 included <strong>SOD1</strong> mutants (20 SOD1, 63 controls), 5 included <strong>FUS</strong> mutants (14 FUS, 55 controls), 3 included <strong>TARDBP</strong> mutants (10 TARDBP, 48 controls) and 2 included <strong>sporadic iPSMNs</strong> (208 sporadic, 56 controls). Controls from each dataset were utilised only if the dataset had samples from the relevant genetic background.</p>
<h4 id="DEG"><a href="#DEG" class="headerlink" title="DEG:"></a>DEG:</h4><p>TARDBP mutants (3,547), FUS (239), C9orf72 (161), and SOD1 (7), sporadic ALS(4)<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020234908792.png" srcset="/img/loading.gif" lazyload><br>Although no genes were significantly changed in expression across all ALS genetic backgrounds, Uroplakin <strong>UPK3BL1</strong> and nuclear pore complex interacting protein <strong>NPIPA8</strong> were changed in the C9orf72, TARDBP and FUS subgroups (Supplementary Fig. 12b).<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020235249322.png" srcset="/img/loading.gif" lazyload></p>
<h4 id="Pearson-correlation-between-ALS-genetic-backgrounds"><a href="#Pearson-correlation-between-ALS-genetic-backgrounds" class="headerlink" title="Pearson correlation between ALS genetic backgrounds"></a>Pearson correlation between ALS genetic backgrounds</h4><p>Correlating transcriptome-wide gene expression changes between ALS genetic backgrounds revealed weak associations, with <strong>the strongest correlation between SOD1 and sporadic lines</strong> (Pearson R &#x3D; + 0.38, p &lt; 2.2 × 10−16) and <strong>the weakest between SOD1 and TARDBP</strong> (R &#x3D; −0.14, p &lt; 2.2 × 10−16; Fig. 3f).<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020235048783.png" srcset="/img/loading.gif" lazyload></p>
<h4 id="ORA-1"><a href="#ORA-1" class="headerlink" title="ORA"></a>ORA</h4><p>Functional enrichment terms enriched in (c) C9orf72, (d) FUS and (e) TARDBP using the hypergeometric test.<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020235504303.png" srcset="/img/loading.gif" lazyload><br>C9orf72 mutants: upregulated genes involved with <strong>p53</strong> (hypergeometric FDR &#x3D; 0.003) and the DNA damage response (FDR &#x3D; 0.02) whilst downregulating cytoskeleton (FDR &#x3D; 0.01) and microtubule genes (FDR &#x3D; 0.02).<br>FUS mutants: upregulation of genes involved with transcription (FDR &#x3D; 1.5 × 10−8) and DNA-binding (FDR &#x3D; 2.4 × 10−8) and downregulation of synaptic signalling genes (FDR &#x3D; 0.02). TARDBP mutants: upregulated neuronal (FDR &#x3D; 7.8 × 10−29) and synaptic genes (FDR &#x3D; 1.2 × 10−32) and downregulated genes involved in the cell cycle (FDR &#x3D; 3.1 × 1030) and RNA splicing (FDR &#x3D; 2 × 10−5, Supplementary Fig. 12c-e).<br>There were no functional terms enriched amongst SOD1 or sporadic ALS differentially expressed genes.</p>
<h4 id="PROGENy-pathway-activities"><a href="#PROGENy-pathway-activities" class="headerlink" title="PROGENy pathway activities"></a>PROGENy pathway activities</h4><p>Apart from SOD1 (NES + 0.33, p &#x3D; 0.16), the p53 pathway activity was significantly increased in each of C9orf72 (NES + 10.9, p &lt; 0.001), TARDBP (NES + 8.6, p &lt; 0.001), sporadic (NES + 4.2, p &lt; 0.001) and FUS (NES + 3.2, p &#x3D; 0.018; Fig. 3g). Examining the other signalling pathways revealed that hypoxia, VEGF and MAPK were also increased across most genetic backgrounds, whilst WNT and PI3K tended to be decreased.</p>
<h4 id="TF-regulon-activity"><a href="#TF-regulon-activity" class="headerlink" title="TF regulon activity"></a>TF regulon activity</h4><p>Significantly increased TP53 activity in C9orf72  (NES + 9.3, p &lt; 0.001), FUS (NES + 4.0, p &#x3D; 0.004) and TARDBP (NES + 2.5, p &#x3D; 0.01) but decreased activity in SOD1 (NES −3.0, p &lt; 0.001) whilst sporadic (NES −0.31, p &#x3D; 0.25) was non-significantly changed.<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241020235343196.png" srcset="/img/loading.gif" lazyload></p>
<h2 id="ALS-post-mortem-tissue-shows-p53-activation"><a href="#ALS-post-mortem-tissue-shows-p53-activation" class="headerlink" title="ALS post-mortem tissue shows p53 activation"></a>ALS <strong>post-mortem tissue</strong> shows p53 activation</h2><p>214 ALS patients and 57 controls</p>
<h3 id="Pan-ALS"><a href="#Pan-ALS" class="headerlink" title="Pan-ALS"></a>Pan-ALS</h3><p><strong>upregulated DEG</strong>:<br>6575 (includes CHIT1, GPNMB and LYZ)<br><strong>downregulated</strong>:<br>7489</p>
<p>As with iPSMNs, in ALS post-mortem spinal cord, the most strongly upregulated PROGENy signalling pathway was <strong>p53</strong> (NES + 5.0, p &lt; 0.001; Fig. 4d). The top individual genes driving p53 pathway activity in ALS post-mortem included FAS, RRM2B, CSTA, ZMAT3, TP53I3, and CDKN1A ( Fig. 4e)<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241021000919713.png" srcset="/img/loading.gif" lazyload></p>
<p>Correlating transcriptome-wide ALS gene expression changes between iPSMNs and post-mortem spinal cord revealed a weak positive correlation (Pearson R &#x3D; + 0.12, p &lt; 2.2 × 10−16; Fig. 4g).<br>These results not only confirm p53-dependent DNA damage response upregulation in ALS but the overlap between iPSMNs and post-mortem provides insight into motor neuron-specific changes that begin early and persist into the later stages of the disease.<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241021001700232.png" srcset="/img/loading.gif" lazyload></p>
<h3 id="Different-ALS-genetic-backgrounds"><a href="#Different-ALS-genetic-backgrounds" class="headerlink" title="Different ALS genetic backgrounds"></a>Different ALS genetic backgrounds</h3><p>In contrast to iPSMNs, comparing genetic subgroups from postmortem spinal cord tissue revealed strongly correlated gene expression changes (R range +0.68 to +0.9) with 1750 overlapping differentially expressed genes between sporadic, C9orf72, SOD1, and FUS subgroups.<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241021001802404.png" srcset="/img/loading.gif" lazyload></p>
<h2 id="TDP-43-pathology-contributes-to-the-DNA-damage-response"><a href="#TDP-43-pathology-contributes-to-the-DNA-damage-response" class="headerlink" title="TDP-43 pathology contributes to the DNA damage response"></a>TDP-43 pathology contributes to the DNA damage response</h2><p>Although TDP-43 pathology (characterised by neuronal TDP-43 nuclear depletion and cytoplasmic accumulation) is observed in 97% of ALS, it is <strong>absent in SOD1 and FUS mutant</strong> cases (termed “<strong>non-TDP-43 ALS</strong>” hereafter).<br>To identify the degree to which genotypes linked to TDP-43 pathology contribute to p53 upregulation in the pan-ALS analyses, we classified ALS samples based on whether their genetic background is associated with TDP-43 pathology.<br>These findings suggest that the p53 signature from the pan-ALS analyses is largely driven by genetic backgrounds associated with TDP-43 proteinopathy.<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241021003134057.png" srcset="/img/loading.gif" lazyload></p>
<p>To discover whether p53 signalling changes are regulated by TDP43, we next examined RNA-seq from FACS-sorted neuronal nuclei into those with and without TDP-43 pathology from FTD-ALS post-mortem brain tissue.<br> Together, these results indicate that both depletion and accumulation of TDP-43 augment p53 activity, suggesting that tight regulation of TDP-43 levels is required to ensure an appropriate DNA damage response.<br><img src="/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/image-20241021003032637.png" srcset="/img/loading.gif" lazyload></p>
<h1 id="Discussion"><a href="#Discussion" class="headerlink" title="Discussion"></a>Discussion</h1><p><strong>PROGENy</strong> and <strong>DoRothEA</strong> weights are based on perturbation experiments that are <strong>not specific to motor neurons</strong>. Their signalling pathways may activate diverse downstream gene expression programmes depending on the cell type and perturbing agent utilised.</p>
<h1 id="Reference"><a href="#Reference" class="headerlink" title="Reference"></a>Reference</h1><ol>
<li>Feldman EL, Goutman SA, Petri S, Mazzini L, Savelieff MG, Shaw PJ, et al. Amyotrophic lateral sclerosis. The Lancet. 2022 Oct;400(10360):1363–80.</li>
</ol>

                
              </div>
            
            <hr/>
            <div>
              <div class="post-metas my-3">
  
  
    <div class="post-meta">
      <i class="iconfont icon-tags"></i>
      
        <a href="/tags/%E6%8A%80%E6%9C%AF/">#技术</a>
      
    </div>
  
</div>


              
  

  <div class="license-box my-3">
    <div class="license-title">
      <div>Integrated transcriptome landscape of ALS identifies genome instability linked to TDP-43 pathology</div>
      <div>https://xuan-98-l.github.io/2024/10/21/ziffIntegratedTranscriptomeLandscape2023a/</div>
    </div>
    <div class="license-meta">
      
        <div class="license-meta-item">
          <div>Author</div>
          <div>Xuan Liu</div>
        </div>
      
      
        <div class="license-meta-item license-meta-date">
          <div>Posted on</div>
          <div>October 21, 2024</div>
        </div>
      
      
      
        <div class="license-meta-item">
          <div>Licensed under</div>
          <div>
            
              
              
                <a target="_blank" href="https://creativecommons.org/licenses/by/4.0/">
                  <span class="hint--top hint--rounded" aria-label="BY - Attribution">
                    <i class="iconfont icon-by"></i>
                  </span>
                </a>
              
            
          </div>
        </div>
      
    </div>
    <div class="license-icon iconfont"></div>
  </div>



              
                <div class="post-prevnext my-3">
                  <article class="post-prev col-6">
                    
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2024/10/16/local-git/" title="git 本地版本管理">
                        <span class="hidden-mobile">git 本地版本管理</span>
                        <span class="visible-mobile">Next</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>

    <div class="side-col d-none d-lg-block col-lg-2">
      
  <aside class="sidebar" style="margin-left: -1rem">
    <div id="toc">
  <p class="toc-header">
    <i class="iconfont icon-list"></i>
    <span>Table of Contents</span>
  </p>
  <div class="toc-body" id="toc-body"></div>
</div>



  </aside>


    </div>
  </div>
</div>





  



  



  



  



  







    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">Search</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v" for="local-search-input">Keyword</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>

    

    
  </main>

  <footer>
    <div class="footer-inner">
  
    <div class="footer-content">
       <a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a> <i class="iconfont icon-love"></i> <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener"><span>Fluid</span></a> 
    </div>
  
  
  
  
</div>

  </footer>

  <!-- Scripts -->
  
  <script  src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://lib.baomitu.com/jquery/3.6.0/jquery.min.js" ></script>
<script  src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js" ></script>
<script  src="/js/events.js" ></script>
<script  src="/js/plugins.js" ></script>


  <script  src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var subtitle = document.getElementById('subtitle');
      if (!subtitle || !typing) {
        return;
      }
      var text = subtitle.getAttribute('data-typed-text');
      
        typing(text);
      
    })(window, document);
  </script>




  
    <script  src="/js/img-lazyload.js" ></script>
  




  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/tocbot/4.18.2/tocbot.min.js', function() {
    var toc = jQuery('#toc');
    if (toc.length === 0 || !window.tocbot) { return; }
    var boardCtn = jQuery('#board-ctn');
    var boardTop = boardCtn.offset().top;

    window.tocbot.init(Object.assign({
      tocSelector     : '#toc-body',
      contentSelector : '.markdown-body',
      linkClass       : 'tocbot-link',
      activeLinkClass : 'tocbot-active-link',
      listClass       : 'tocbot-list',
      isCollapsedClass: 'tocbot-is-collapsed',
      collapsibleClass: 'tocbot-is-collapsible',
      scrollSmooth    : true,
      includeTitleTags: true,
      headingsOffset  : -boardTop,
    }, CONFIG.toc));
    if (toc.find('.toc-list-item').length > 0) {
      toc.css('visibility', 'visible');
    }

    Fluid.events.registerRefreshCallback(function() {
      if ('tocbot' in window) {
        tocbot.refresh();
        var toc = jQuery('#toc');
        if (toc.length === 0 || !tocbot) {
          return;
        }
        if (toc.find('.toc-list-item').length > 0) {
          toc.css('visibility', 'visible');
        }
      }
    });
  });
</script>


  <script src=https://lib.baomitu.com/clipboard.js/2.0.11/clipboard.min.js></script>

  <script>Fluid.plugins.codeWidget();</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js', function() {
    window.anchors.options = {
      placement: CONFIG.anchorjs.placement,
      visible  : CONFIG.anchorjs.visible
    };
    if (CONFIG.anchorjs.icon) {
      window.anchors.options.icon = CONFIG.anchorjs.icon;
    }
    var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
    var res = [];
    for (var item of el) {
      res.push('.markdown-body > ' + item.trim());
    }
    if (CONFIG.anchorjs.placement === 'left') {
      window.anchors.options.class = 'anchorjs-link-left';
    }
    window.anchors.add(res.join(', '));

    Fluid.events.registerRefreshCallback(function() {
      if ('anchors' in window) {
        anchors.removeAll();
        var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
        var res = [];
        for (var item of el) {
          res.push('.markdown-body > ' + item.trim());
        }
        if (CONFIG.anchorjs.placement === 'left') {
          anchors.options.class = 'anchorjs-link-left';
        }
        anchors.add(res.join(', '));
      }
    });
  });
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js', function() {
    Fluid.plugins.fancyBox();
  });
</script>


  <script>Fluid.plugins.imageCaption();</script>

  <script  src="/js/local-search.js" ></script>





<!-- 主题的启动项，将它保持在最底部 -->
<!-- the boot of the theme, keep it at the bottom -->
<script  src="/js/boot.js" ></script>


  

  <noscript>
    <div class="noscript-warning">Blog works best with JavaScript enabled</div>
  </noscript>
</body>
</html>
