

<!DOCTYPE html>
<html lang="en" data-default-color-scheme=auto>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/fluid.png">
  <link rel="icon" href="/img/fluid.png">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=5.0, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="author" content="Xuan Liu">
  <meta name="keywords" content="">
  
    <meta name="description" content="Monocle2的学习笔记及代码分享">
<meta property="og:type" content="article">
<meta property="og:title" content="Monocle2">
<meta property="og:url" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/index.html">
<meta property="og:site_name" content="Hexo">
<meta property="og:description" content="Monocle2的学习笔记及代码分享">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/scRNA_pseudotime_estimation_problem.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/plot1.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/banana.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/heatmap1.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/algorithm1.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/p1.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/p2.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/p3.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/p4.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/vis1.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/vis2.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/vis3.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/vis4.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/high.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/velocity.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/diffusion1.png">
<meta property="og:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/diffusion2.png">
<meta property="article:published_time" content="2023-05-19T03:31:28.000Z">
<meta property="article:modified_time" content="2024-08-20T00:27:29.144Z">
<meta property="article:author" content="Xuan Liu">
<meta property="article:tag" content="技术">
<meta property="article:tag" content="个人成长">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image" content="https://xuan-98-l.github.io/2023/05/19/Monocle2/scRNA_pseudotime_estimation_problem.png">
  
  
  
  <title>Monocle2 - Hexo</title>

  <link  rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css" />



  <link  rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css" />

  <link  rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css" />



<!-- 主题依赖的图标库，不要自行修改 -->
<!-- Do not modify the link that theme dependent icons -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css">


<link  rel="stylesheet" href="/css/main.css" />


  <link id="highlight-css" rel="stylesheet" href="/css/highlight.css" />
  
    <link id="highlight-css-dark" rel="stylesheet" href="/css/highlight-dark.css" />
  




  <script id="fluid-configs">
    var Fluid = window.Fluid || {};
    Fluid.ctx = Object.assign({}, Fluid.ctx)
    var CONFIG = {"hostname":"xuan-98-l.github.io","root":"/","version":"1.9.4","typing":{"enable":true,"typeSpeed":70,"cursorChar":"_","loop":false,"scope":[]},"anchorjs":{"enable":true,"element":"h1,h2,h3,h4,h5,h6","placement":"left","visible":"hover","icon":""},"progressbar":{"enable":true,"height_px":3,"color":"#29d","options":{"showSpinner":false,"trickleSpeed":100}},"code_language":{"enable":true,"default":"TEXT"},"copy_btn":true,"image_caption":{"enable":true},"image_zoom":{"enable":true,"img_url_replace":["",""]},"toc":{"enable":true,"placement":"right","headingSelector":"h1,h2,h3,h4,h5,h6","collapseDepth":0},"lazyload":{"enable":true,"loading_img":"/img/loading.gif","onlypost":false,"offset_factor":2},"web_analytics":{"enable":false,"follow_dnt":true,"baidu":null,"google":null,"gtag":null,"tencent":{"sid":null,"cid":null},"woyaola":null,"cnzz":null,"leancloud":{"app_id":null,"app_key":null,"server_url":null,"path":"window.location.pathname","ignore_local":false}},"search_path":"/local-search.xml"};

    if (CONFIG.web_analytics.follow_dnt) {
      var dntVal = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
      Fluid.ctx.dnt = dntVal && (dntVal.startsWith('1') || dntVal.startsWith('yes') || dntVal.startsWith('on'));
    }
  </script>
  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
  


  
<meta name="generator" content="Hexo 6.3.0"></head>


<body>
  

  <header>
    

<div class="header-inner" style="height: 70vh;">
  <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand" href="/">
      <strong>Xuan</strong>
    </a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/">
                <i class="iconfont icon-home-fill"></i>
                <span>Home</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/">
                <i class="iconfont icon-archive-fill"></i>
                <span>Archives</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/">
                <i class="iconfont icon-category-fill"></i>
                <span>Categories</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/">
                <i class="iconfont icon-tags-fill"></i>
                <span>Tags</span>
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/">
                <i class="iconfont icon-user-fill"></i>
                <span>About</span>
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search">
              <i class="iconfont icon-search"></i>
            </a>
          </li>
          
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle">
              <i class="iconfont icon-dark" id="color-toggle-icon"></i>
            </a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

  

<div id="banner" class="banner" parallax=true
     style="background: url('/img/default.png') no-repeat center center; background-size: cover;">
  <div class="full-bg-img">
    <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
      <div class="banner-text text-center fade-in-up">
        <div class="h2">
          
            <span id="subtitle" data-typed-text="Monocle2"></span>
          
        </div>

        
          
  <div class="mt-3">
    
    
      <span class="post-meta">
        <i class="iconfont icon-date-fill" aria-hidden="true"></i>
        <time datetime="2023-05-19 13:31" pubdate>
          May 19, 2023 pm
        </time>
      </span>
    
  </div>

  <div class="mt-1">
    
      <span class="post-meta mr-2">
        <i class="iconfont icon-chart"></i>
        
          8.2k words
        
      </span>
    

    
      <span class="post-meta mr-2">
        <i class="iconfont icon-clock-fill"></i>
        
        
        
          69 mins
        
      </span>
    

    
    
  </div>


        
      </div>

      
    </div>
  </div>
</div>

</div>

  </header>

  <main>
    
      

<div class="container-fluid nopadding-x">
  <div class="row nomargin-x">
    <div class="side-col d-none d-lg-block col-lg-2">
      

    </div>

    <div class="col-lg-8 nopadding-x-md">
      <div class="container nopadding-x-md" id="board-ctn">
        <div id="board">
          <article class="post-content mx-auto">
            <!-- SEO header -->
            <h1 style="display: none">Monocle2</h1>
            
            
              <div class="markdown-body">
                
                <p>Monocle2的学习笔记及代码分享</p>
<span id="more"></span>
<h2 id="背景介绍"><a href="#背景介绍" class="headerlink" title="背景介绍"></a>背景介绍</h2><p>在研究胚胎发育，细胞转化以及干细胞在再生和疾病中的功能等领域，通过<a target="_blank" rel="noopener" href="https://www.molcells.org/journal/view.html?doi=10.14348/molcells.2019.0006">谱系追踪</a>来研究单个细胞及其后代的命运是常见的研究方法。传统的谱系追踪涉及用染料显微注射到单个细胞或细胞群中以观察细胞迁移，或用利用Cre-LoxP 位点特异性重组系统对细胞命运进行追踪展示等基于成像的方法（Imageing-gased approaches）。</p>
<p>在NGS测序出现后，使用计算手段来追踪或重建细胞谱系对谱系追踪领域影响巨大。基于scRNA-seq，还可以在进行建立细胞谱系的同时分析基因表达模式。</p>
<p>由于细胞分化或细胞成熟等细胞进程在生物进程中并不总是以一种分散的状态，比如clustering，进行的，而更多是一种动态的，连续的发育状态。因此，很多<a target="_blank" rel="noopener" href="https://github.com/agitter/single-cell-pseudotime">计算方法</a>致力于将细胞投射到一个连续路径上（Pseudotime）来展示细胞的发育轨迹。</p>
<p>下图为拟时序分析的简单<a target="_blank" rel="noopener" href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005212">示例图</a><br><img src="/2023/05/19/Monocle2/scRNA_pseudotime_estimation_problem.png" srcset="/img/loading.gif" lazyload></p>
<p><img src="/2023/05/19/Monocle2/plot1.png" srcset="/img/loading.gif" lazyload></p>
<h2 id="Monocle介绍"><a href="#Monocle介绍" class="headerlink" title="Monocle介绍"></a>Monocle介绍</h2><h3 id="软件目的"><a href="#软件目的" class="headerlink" title="软件目的:"></a>软件目的:</h3><p>1: 对从细胞身份转化的过程进行建模（从一个clusters进展到另一个cluster），研究分化过程。</p>
<p>2: 可以使用轨迹之间的差异来识别与轨迹转变相关的基因</p>
<p>注意：推断的轨迹过程有时不一定反应真实的生物学过程，需要结合其他生物学先验知识和证据来对轨迹进行解释。</p>
<h3 id="研发历史"><a href="#研发历史" class="headerlink" title="研发历史"></a>研发历史</h3><h4 id="2014年-Nature-Biotechnology"><a href="#2014年-Nature-Biotechnology" class="headerlink" title="2014年 Nature Biotechnology"></a><a target="_blank" rel="noopener" href="https://doi.org/10.1038/nbt.2859">2014年 Nature Biotechnology</a></h4><p>monocle的概念版本发表在2014年，这个时间点，基于微流控的高通量的单细胞测序技术，比如droplet-based的scRNA-seq技术都还没有发表，商业化的10X单细胞测序也还未出现，此时的单细胞测序还是通过显微成像或者流式细胞技术进行细胞水平的研究。因此这个版本的Monocle针对的其实是低细胞数的轨迹（trajectory）分析。虽然和现在的10X单细胞测序动辄能测上万细胞的通量相比，当时用作分析的细胞数也就是几百个，相比之下低了太多，但就算如此，bulk RNA seq转录组测序的差异表达基因算法仍然不适用与单细胞测序。因为Bulk RNA-seq测序数据的差异分析通常采用基于平均值或方差的假设检验方法，无法捕捉到单个细胞水平的表达动态。</p>
<p>Monocle1，Monocle的初始版本，是第一个发布的单细胞RNA测序数据分析软件包之一，它认为就算单细胞测序是在同一时间捕获到的细胞群体，它也可能包括许多不同的中间分化状态，传统转录组那种只考虑其平均值的算法会掩盖在单个细胞中发生的趋势。</p>
<p><img src="/2023/05/19/Monocle2/banana.png" srcset="/img/loading.gif" lazyload></p>
<p>Monocle基于一个基本假设：scRNA - Seq中的所有细胞存在潜在的时间顺序，由于分化不同步使得细胞的基因表达水平存在很强的差异性，那就可以根据这个潜在的动态过程对细胞进行排序。Monocle的算法源自先前的基于bulk RNASeq重构样本间的时间顺序的算法，但将其扩展到单细胞层面，并允许单个祖细胞类型存在多个细胞命运。</p>
<p>Monocle引入了一个概念： “pseudotime”， 即通过生物过程进展的定量测量，在’pseudotime拟时间’中排序单细胞表达谱。或者说是细胞在发育过程中的时间顺序。monocle1可以根据基因表达数据生成拓扑图，确定基因表达水平的变化趋势，并可视化地展示出细胞发育轨迹，从而更容易理解和解释单细胞RNA-seq数据的结果。</p>
<p>热图展示了Monocle找到的差异基因(行)，细胞(列)按拟时间顺序排列。<br><img src="/2023/05/19/Monocle2/heatmap1.png" srcset="/img/loading.gif" lazyload></p>
<p>工作原理图<br><img src="/2023/05/19/Monocle2/algorithm1.png" srcset="/img/loading.gif" lazyload></p>
<h4 id="2017年-Nature-Methods-1月"><a href="#2017年-Nature-Methods-1月" class="headerlink" title="2017年 Nature Methods (1月)"></a><a target="_blank" rel="noopener" href="https://doi.org/10.1038/nmeth.4150">2017年 Nature Methods (1月)</a></h4><p>随着测序技术的发展，高通量的单细胞测序技术产生了，单细胞测序可以检测到更多的细胞和更多的基因表达水平。为了适应这种变化，monocle2，在2017年被发表在nature methods上。这个版本的monocle对之前算法进行了进一步优化，增加了用于分支分析的BEAM算法。</p>
<p><strong>Branched expression analysis modeling (BEAM)</strong></p>
<p>为了以稳健的方式从学习到的轨迹中检测细胞命运依赖基因，作者开发了BEAM，一种广义线性模型，用于找到分支之间不同的所有基因，（其实就是一种差异分析方法）。什么是分支呢，就是当发育过程中，细胞执行不同的基因表达程序时，这个时候就是细胞做出命运抉择的时刻。当细胞做出命运选择时，分支就会出现在轨迹中: 一个发育谱系沿着一条路径前进，而另一个谱系产生第二条路径。BEAM算法就可以用来识别在每个谱系分岔过程中表现出显著差异的基因，帮助确定在复杂的生物学过程中细胞命运决定的时刻。</p>
<p>BEAM方法接受一个已通过orderCells函数排序的CellDataSet以及发育轨迹中分支点的名称作为输入，返回每个基因的显着性得分表。得分显着的基因在其表达中被认为是分支依赖性的。</p>
<p>下图展示了’pseudotime拟时间”排序后的细胞轨迹，可以看到上面有黑色的圆圈，是分支点1， branch_point 1，也就是细胞分化细胞命运决定的时刻。<br><img src="/2023/05/19/Monocle2/p1.png" srcset="/img/loading.gif" lazyload></p>
<p>图中简单展示了Beam算法的原理，它使用广义线性模型对已知分支值的数据进行回归(alternative model)，为每个分支拟合单独的曲线。它还通过对所有数据拟合一条曲线来执行分支指派未知的回归(零模型)，然后通过似然比检验比较这些模型。 判断当细胞从树的左上角(E14.5)的早期发育阶段通过分支时，哪些基因发生了变化？哪些基因在分支之间差异表达？<br><img src="/2023/05/19/Monocle2/p2.png" srcset="/img/loading.gif" lazyload></p>
<p>可以看到分化的关键基因（marker基因）Pdpn and Sftpb在明显属于随不同branch变化的基因， 两个分支之间呈现出了显著差异的趋势，但housekeeping genes (Hprt and Pgk1)并未有明显差异。<br><img src="/2023/05/19/Monocle2/p3.png" srcset="/img/loading.gif" lazyload></p>
<h4 id="2017年-Nature-Methods-8月"><a href="#2017年-Nature-Methods-8月" class="headerlink" title="2017年 Nature Methods (8月)"></a><a target="_blank" rel="noopener" href="https://doi.org/10.1038/nmeth.4402">2017年 Nature Methods (8月)</a></h4><p>这篇文献主要讲述了Monocle2的工作流程，介绍了如何将反向图嵌入引入细胞轨迹分析，并改为使用DDRTree的算法做降维。monocle2的稳定版本从此开发出来，它也成为了单细胞领域目前应用最广泛的轨迹推断方法。</p>
<p>monocle2的流程图<br><img src="/2023/05/19/Monocle2/p4.png" srcset="/img/loading.gif" lazyload></p>
<p>关键步骤：</p>
<ol>
<li>选择定义过程的基因: 在单细胞测序中，会出现很多低水平表达的基因，造成一个噪声的影响，因此我们在分析的时候不能选择使用全部的基因分析。推断单细胞的轨迹是一个机器学习的问题，轨迹推断的第一步就是选择monocle使用哪些基因作为轨迹推断的关键基因。也就是机器学习中的特征选择问题。</li>
</ol>
<p>Monocle2首先使用了”dpFeature”方法确认与生物过程相关的基因（这些基因在下游分析时被称为ordering gene），注意一下，这里与BEAM这个寻找分叉点相关基因的算法不同，dpFeatures找到的是全局的拟时相关基因，也就是拟时起点和终点相关的基因。</p>
<ol start="2">
<li><p>每个细胞被表示为高维空间(x)中的一个点，其中每个维度对应一个基因的表达水平。数据从高维空间(X)被投影到低维空间(z)上，使用k - means聚类算法找到这些细胞的分群中心，然后通过反向图嵌入（默认是DDRTree）的流形学习算法，找到高维基因表达空间与低维空间的映射关系并构建最小生成树，将细胞投影到生成树上，并将找到的中心进行连接，构建初步的树形结构。</p>
</li>
<li><p>基于新的细胞位置，更新细胞的分群中心和轨迹结构，将二维空间中的树（二维轨迹），重新投射到高维空间。迭代此过程，直至收敛（即获得稳定的树形结构，细胞轨迹可以反应原始数据的特征）。</p>
</li>
</ol>
<p>这里提到了两个关键词，dpFeature, 用于确认与生物过程相关的基因，也就是ordering gene。以及DDRTree，在monocle的reduceDimension函数中默认使用，用来推断基因表达动态变化，这两个算法是monocle2的主打点。</p>
<p>其实目前monocle已经更新到3代了，但考虑到目前发表的文章还是使用monocle2的比较多，而且monocle3的下游可视化方式还有待开发，我们仍然使用monocle2进行讲解。</p>
<h3 id="应用场景"><a href="#应用场景" class="headerlink" title="应用场景"></a>应用场景</h3><ul>
<li><p>挖掘细胞分化的关键基因：<br>在分化或发育研究中，可通过拟时序分析关键的分化节点，挖掘细胞不同分化路径命运选择的推动基因</p>
</li>
<li><p>揭示细胞发育轨迹：<br>对于未知的细胞间或细胞亚群，通过拟时序分析揭示细胞分化轨迹，推测干细胞在发育过程的分化轨迹或某类细胞的分化来源</p>
</li>
<li><p>揭示细胞动态变化：<br>在肿瘤或免疫研究中，通过拟时序分析追踪细胞的动态变化，推断细胞亚型的演变轨迹或细胞凋亡路径</p>
</li>
</ul>
<h3 id="代码关键步骤讲解"><a href="#代码关键步骤讲解" class="headerlink" title="代码关键步骤讲解"></a>代码关键步骤讲解</h3><p>monocle2 <a target="_blank" rel="noopener" href="https://github.com/cole-trapnell-lab/monocle-release/tree/master">github</a><br>monocle <a target="_blank" rel="noopener" href="https://rdrr.io/bioc/monocle/">R包说明文档</a></p>
<ol>
<li><p>CellDataSet对象构建与预处理</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#可选方案: 从表达量矩阵和细胞、基因的表型文件构建CellDataSet对象</span><br><br><span class="hljs-comment">#1. expr_matrix</span><br><span class="hljs-comment">#行为基因，列为细胞的表达量矩阵</span><br><span class="hljs-comment">#2. phenoData</span><br><span class="hljs-comment">#对应的细胞的标签信息（barcodes)</span><br><span class="hljs-comment">#3. featureData</span><br><span class="hljs-comment">#对应的基因的标签信息，其中必须包含“gene_short_name”列</span><br><br><span class="hljs-comment">##pd, phenoData</span><br>pd <span class="hljs-operator">&lt;-</span> new<span class="hljs-punctuation">(</span><span class="hljs-string">&quot;AnnotatedDataFrame&quot;</span><span class="hljs-punctuation">,</span> data <span class="hljs-operator">=</span> seurat_ob<span class="hljs-operator">@</span>meta.data<span class="hljs-punctuation">)</span> <br><span class="hljs-comment">##fd, featureData</span><br>gene_annotation<span class="hljs-operator">=</span>data.frame<span class="hljs-punctuation">(</span>gene_short_name <span class="hljs-operator">=</span> rownames<span class="hljs-punctuation">(</span>seurat_ob<span class="hljs-operator">@</span>assays<span class="hljs-operator">$</span>RNA<span class="hljs-operator">@</span>counts<span class="hljs-punctuation">)</span><span class="hljs-punctuation">,</span><br>                           stringsAsFactors<span class="hljs-operator">=</span><span class="hljs-built_in">F</span><span class="hljs-punctuation">)</span><br>rownames<span class="hljs-punctuation">(</span>gene_annotation<span class="hljs-punctuation">)</span><span class="hljs-operator">&lt;-</span>gene_annotation<span class="hljs-operator">$</span>gene_short_name<br>fd <span class="hljs-operator">&lt;-</span> new<span class="hljs-punctuation">(</span><span class="hljs-string">&quot;AnnotatedDataFrame&quot;</span><span class="hljs-punctuation">,</span> data <span class="hljs-operator">=</span> gene_annotation<span class="hljs-punctuation">)</span>    <br><br>gbm_cds <span class="hljs-operator">&lt;-</span> newCellDataSet<span class="hljs-punctuation">(</span>GetAssayData<span class="hljs-punctuation">(</span>seurat_ob<span class="hljs-punctuation">,</span>slot<span class="hljs-operator">=</span><span class="hljs-string">&quot;counts&quot;</span><span class="hljs-punctuation">,</span>assay<span class="hljs-operator">=</span><span class="hljs-string">&quot;RNA&quot;</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">,</span><br>                       phenoData <span class="hljs-operator">=</span> pd<span class="hljs-punctuation">,</span><br>                       featureData <span class="hljs-operator">=</span> fd<span class="hljs-punctuation">,</span><br>                    expressionFamily<span class="hljs-operator">=</span>negbinomial.size<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment">#推荐方案</span><br><span class="hljs-comment">#从Seurat聚类结果构建CellDataSet对象</span><br><br>cds <span class="hljs-operator">&lt;-</span> as.CellDataSet<span class="hljs-punctuation">(</span>seurat_ob<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment">#size facotr帮助我们标准化细胞之间的mRNA的差异。</span><br><span class="hljs-comment">#离散度值可以帮助我们进行后续的差异分析。</span><br>cds <span class="hljs-operator">&lt;-</span> estimateSizeFactors<span class="hljs-punctuation">(</span>cds<span class="hljs-punctuation">)</span><br>cds <span class="hljs-operator">&lt;-</span> estimateDispersions<span class="hljs-punctuation">(</span>cds<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment">#过滤低质量细胞(保留至少在1个细胞中counts大于1的基因）</span><br>cds <span class="hljs-operator">=</span>  detectGenes<span class="hljs-punctuation">(</span>cds<span class="hljs-punctuation">,</span>min_expr <span class="hljs-operator">=</span> <span class="hljs-number">1</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment">#保留至少在10个细胞中表达的基因</span><br>expressed_genes <span class="hljs-operator">=</span> row.names<span class="hljs-punctuation">(</span>subset<span class="hljs-punctuation">(</span>fData<span class="hljs-punctuation">(</span>cds<span class="hljs-punctuation">)</span><span class="hljs-punctuation">,</span>num_cells_expressed<span class="hljs-operator">&gt;</span><span class="hljs-number">10</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>
</li>
<li><p>寻找生物学过程相关的基因</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#推荐方案: 使用monocle的dpFeature方法</span><br><span class="hljs-comment">#differentialGeneTest函数即为monocle的dpFeature方法，用于确认与生物过程相关的基因（即与clusters分群相关的差异基因）</span><br><span class="hljs-comment">#The group design to find ordering genes, 这里使用的是seurat的clusters</span><br>clustering_DEGs <span class="hljs-operator">=</span> differentialGeneTest<span class="hljs-punctuation">(</span>cds<span class="hljs-punctuation">[</span>expressed_genes<span class="hljs-punctuation">,</span><span class="hljs-punctuation">]</span><span class="hljs-punctuation">,</span>fullModelFormulaStr <span class="hljs-operator">=</span><span class="hljs-string">&quot;~clusters&quot;</span><span class="hljs-punctuation">,</span>cores <span class="hljs-operator">=</span> <span class="hljs-number">4</span><span class="hljs-punctuation">)</span><br>featureData<span class="hljs-punctuation">(</span>cds<span class="hljs-punctuation">)</span><span class="hljs-operator">@</span>data<span class="hljs-punctuation">[</span>rownames<span class="hljs-punctuation">(</span>clustering_DEGs<span class="hljs-punctuation">)</span><span class="hljs-punctuation">,</span><span class="hljs-string">&quot;pval&quot;</span><span class="hljs-punctuation">]</span><span class="hljs-operator">=</span>clustering_DEGs<span class="hljs-operator">$</span>pval<br>featureData<span class="hljs-punctuation">(</span>cds<span class="hljs-punctuation">)</span><span class="hljs-operator">@</span>data<span class="hljs-punctuation">[</span>rownames<span class="hljs-punctuation">(</span>clustering_DEGs<span class="hljs-punctuation">)</span><span class="hljs-punctuation">,</span><span class="hljs-string">&quot;qval&quot;</span><span class="hljs-punctuation">]</span><span class="hljs-operator">=</span>clustering_DEGs<span class="hljs-operator">$</span>qval<br><br><span class="hljs-comment">#根据qval筛选ordering gene</span><br>ordering_genes <span class="hljs-operator">&lt;-</span> row.names <span class="hljs-punctuation">(</span>subset<span class="hljs-punctuation">(</span>clustering_DEGs<span class="hljs-punctuation">,</span> qval <span class="hljs-operator">&lt;</span> <span class="hljs-number">0.01</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span><br><br><span class="hljs-comment">#一旦我们有用于Ordering的基因ID列表，我们就需要将它们设置在cds对象中</span><br>gbm_cds <span class="hljs-operator">=</span> setOrderingFilter<span class="hljs-punctuation">(</span>cds<span class="hljs-punctuation">,</span>ordering_genes <span class="hljs-operator">=</span> ordering_genes<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment">#可以使用fData(gbm_cds)查看，基因矩阵中增加了一列use_for_ordering列</span><br>plot_ordering_genes<span class="hljs-punctuation">(</span>gbm_cds<span class="hljs-punctuation">)</span><br><br><span class="hljs-comment">#可选方案:选择方差最大的基因</span><br>disp_table <span class="hljs-operator">&lt;-</span> dispersionTable<span class="hljs-punctuation">(</span>cds<span class="hljs-punctuation">)</span><br>ordering_genes <span class="hljs-operator">&lt;-</span> subset<span class="hljs-punctuation">(</span>disp_table<span class="hljs-punctuation">,</span><br>                         mean_expression <span class="hljs-operator">&gt;=</span> <span class="hljs-number">0.5</span> <span class="hljs-operator">&amp;</span><br>                           dispersion_empirical <span class="hljs-operator">&gt;=</span> <span class="hljs-number">1</span> <span class="hljs-operator">*</span> dispersion_fit<span class="hljs-punctuation">)</span><span class="hljs-operator">$</span>gene_id<br><span class="hljs-comment">#将ordering gene设置在cds对象</span><br>gbm_cds <span class="hljs-operator">&lt;-</span> setOrderingFilter<span class="hljs-punctuation">(</span>cds<span class="hljs-punctuation">,</span> ordering_genes<span class="hljs-punctuation">)</span><br>plot_ordering_genes<span class="hljs-punctuation">(</span>gbm_cds<span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>
</li>
<li><p>构建生成树</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><code class="hljs R"><span class="hljs-comment">#the dependent package irlba should not be less than 2.3.2, or the following step will fail</span><br><span class="hljs-comment">#if the size of cells is big.</span><br>gbm_cds <span class="hljs-operator">=</span> reduceDimension<span class="hljs-punctuation">(</span>gbm_cds<span class="hljs-punctuation">,</span>max_components <span class="hljs-operator">=</span> <span class="hljs-number">2</span><span class="hljs-punctuation">,</span>verbose <span class="hljs-operator">=</span> <span class="hljs-built_in">T</span><span class="hljs-punctuation">)</span><br>gbm_cds <span class="hljs-operator">=</span> orderCells<span class="hljs-punctuation">(</span>gbm_cds<span class="hljs-punctuation">,</span>reverse <span class="hljs-operator">=</span> <span class="hljs-built_in">F</span><span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure>
</li>
<li><p>可视化结果展示<br>轨迹图<br><img src="/2023/05/19/Monocle2/vis1.png" srcset="/img/loading.gif" lazyload alt="轨迹图"><br>基因表达量<br><img src="/2023/05/19/Monocle2/vis2.png" srcset="/img/loading.gif" lazyload alt="基因表达量"><br>动力学趋势图<br><img src="/2023/05/19/Monocle2/vis3.png" srcset="/img/loading.gif" lazyload alt="动力学趋势图"><br>热图<br><img src="/2023/05/19/Monocle2/vis4.png" srcset="/img/loading.gif" lazyload alt="热图"></p>
</li>
<li><p>进阶练习：将基因表达量映射到拟时序轨迹上<br><img src="/2023/05/19/Monocle2/high.png" srcset="/img/loading.gif" lazyload></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><code class="hljs R">Dims <span class="hljs-operator">=</span> as.data.frame<span class="hljs-punctuation">(</span>t<span class="hljs-punctuation">(</span>monocle<span class="hljs-operator">::</span>reducedDimS<span class="hljs-punctuation">(</span>gbm_cds<span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">%&gt;%</span> rownames_to_column<span class="hljs-punctuation">(</span><span class="hljs-string">&quot;Barcodes&quot;</span><span class="hljs-punctuation">)</span><br>colnames<span class="hljs-punctuation">(</span>Dims<span class="hljs-punctuation">)</span><span class="hljs-punctuation">[</span><span class="hljs-number">2</span><span class="hljs-punctuation">]</span> <span class="hljs-operator">=</span> <span class="hljs-string">&quot;Component1&quot;</span><br>colnames<span class="hljs-punctuation">(</span>Dims<span class="hljs-punctuation">)</span><span class="hljs-punctuation">[</span><span class="hljs-number">3</span><span class="hljs-punctuation">]</span> <span class="hljs-operator">=</span> <span class="hljs-string">&quot;Component2&quot;</span><br><br>i<span class="hljs-operator">=</span> to_be_tested<span class="hljs-punctuation">[</span><span class="hljs-number">1</span><span class="hljs-punctuation">]</span><br>data <span class="hljs-operator">=</span> FetchData<span class="hljs-punctuation">(</span>object <span class="hljs-operator">=</span> seurat_ob<span class="hljs-punctuation">,</span> vars <span class="hljs-operator">=</span> <span class="hljs-built_in">c</span><span class="hljs-punctuation">(</span><span class="hljs-string">&quot;clusters&quot;</span><span class="hljs-punctuation">,</span><span class="hljs-built_in">as.character</span><span class="hljs-punctuation">(</span>i<span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">%&gt;%</span> rownames_to_column<span class="hljs-punctuation">(</span><span class="hljs-string">&quot;Barcodes&quot;</span><span class="hljs-punctuation">)</span><br><br><br>data_df <span class="hljs-operator">&lt;-</span> merge<span class="hljs-punctuation">(</span>data<span class="hljs-punctuation">,</span>Dims<span class="hljs-punctuation">,</span> by<span class="hljs-operator">=</span><span class="hljs-string">&quot;Barcodes&quot;</span><span class="hljs-punctuation">)</span><br><br>g <span class="hljs-operator">&lt;-</span> ggplot<span class="hljs-punctuation">(</span>data<span class="hljs-operator">=</span>data_df<span class="hljs-punctuation">,</span> aes<span class="hljs-punctuation">(</span>x<span class="hljs-operator">=</span>Component1<span class="hljs-punctuation">,</span> y<span class="hljs-operator">=</span>Component2<span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span> <span class="hljs-operator">+</span> geom_point<span class="hljs-punctuation">(</span>aes<span class="hljs-punctuation">(</span>color<span class="hljs-operator">=</span>data_df<span class="hljs-punctuation">[[</span>i<span class="hljs-punctuation">]</span><span class="hljs-punctuation">]</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">,</span> size<span class="hljs-operator">=</span><span class="hljs-number">1.5</span><span class="hljs-punctuation">,</span> na.rm <span class="hljs-operator">=</span> <span class="hljs-literal">TRUE</span><span class="hljs-punctuation">)</span><span class="hljs-operator">+</span>theme_classic<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span><span class="hljs-operator">+</span> scale_color_gradientn<span class="hljs-punctuation">(</span>colours <span class="hljs-operator">=</span> <span class="hljs-built_in">c</span><span class="hljs-punctuation">(</span><span class="hljs-string">&quot;grey&quot;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&quot;yellow&quot;</span><span class="hljs-punctuation">,</span> <span class="hljs-string">&quot;red&quot;</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span><span class="hljs-operator">+</span>theme<span class="hljs-punctuation">(</span>legend.text <span class="hljs-operator">=</span> element_text<span class="hljs-punctuation">(</span>size <span class="hljs-operator">=</span> <span class="hljs-number">10</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">,</span>legend.title <span class="hljs-operator">=</span> element_blank<span class="hljs-punctuation">(</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">,</span>plot.title <span class="hljs-operator">=</span> element_text<span class="hljs-punctuation">(</span>hjust <span class="hljs-operator">=</span> <span class="hljs-number">0.5</span><span class="hljs-punctuation">)</span><span class="hljs-punctuation">)</span><span class="hljs-operator">+</span>labs<span class="hljs-punctuation">(</span>title <span class="hljs-operator">=</span> i<span class="hljs-punctuation">)</span><br></code></pre></td></tr></table></figure></li>
</ol>
<h2 id="其他拟时序软件介绍"><a href="#其他拟时序软件介绍" class="headerlink" title="其他拟时序软件介绍"></a>其他拟时序软件介绍</h2><h3 id="RNA-velocity-RNA速率"><a href="#RNA-velocity-RNA速率" class="headerlink" title="RNA velocity (RNA速率)"></a>RNA velocity (RNA速率)</h3><p>分子生物学上我们都知道，刚转录出的mRNA包含外显子和内含子，经过剪接切除内含子后，得到用于编码蛋白的成熟mRNA，最后经过降解消除。在单细胞测序中，可以很容易观测到剪接的发生，例如跨外显子&#x2F;内含子的reads可表示未剪接的pre-mRNA，跨外显子&#x2F;外显子的reads可表示剪接的成熟mRNA，通过Cellranger软件，也就是在下机数据与基因组比对的初始处理步骤中，可针对所有基因的每个细胞产生剪接和非剪接RNA数量进行计数。如果转录速率恒定，那RNA的丰度（也就是RNA velocity）只由未成熟转向成熟，以及成熟后的降解两个因素所决定。通过计算剪接和未剪接的mRNA分子数，我们就可以间接的估算基因的转录速率。对于某个基因来说，如果它新生的RNA”耗尽”了，就表示这个基因的表达被下调，而未成熟mRNA的积累表示这个基因的表达被上调。</p>
<p>如图所示，RNA velocity的基本原理：先是计算所有基因的每个细胞的剪接和未剪接计数，然后推断合适的转录动力学模型。通过计算未剪接转录本和剪接转录本之间的比率可以推断细胞命运的状态(过渡与稳定)和方向性(轨迹)。<br><img src="/2023/05/19/Monocle2/velocity.png" srcset="/img/loading.gif" lazyload><br>使用RNA速率分析，就可以计算细胞内mRNA剪接前后比例来估计RNA丰度随时间的变化，推断转录事件的时间顺序，估算RNA相关的转录剪切降解速率，在<strong>无需先验知识</strong>的情况下，推断细胞的下一个可能的分化方向。</p>
<p>RNA velocity可以揭示单细胞基因表达在时间尺度上的动态变化，这种动态变化与人类和其他哺乳动物的发育、再生和反应过程相匹配。它是一种局部速度向量，可用于模型定型、命运选择和体内转录的精确动力学研究，能够详细研究复杂组织和器官的动态过程，并将极大地促进人类胚胎的谱系分析。</p>
<p>RNA velocity的方法的具体实现通常是通过velocyto和scVelo，由于scVelo软件包的维护做的比较好，并且引入了更优化的模型，基于似然的动力学模型，所以scVelo会更常用一些。由于RNA velocity的环境配置比较复杂，仅做原理讲解，不做实操展示。</p>
<p>应用场景：</p>
<ul>
<li><p>动力学研究：<br>模型定型、命运选择和体内转录的精确动力学研究</p>
</li>
<li><p>揭示细胞动态变化：<br>能够详细研究复杂组织和器官的动态过程，</p>
</li>
<li><p>谱系分析：<br>人类胚胎的谱系分析</p>
</li>
</ul>
<h3 id="Diffusion-map"><a href="#Diffusion-map" class="headerlink" title="Diffusion map"></a>Diffusion map</h3><p>回忆一下之前谈到的monocle的原理，monocle2依赖differentialGeneTest方法，需要现有一个分组的信息才能进行轨迹构建。但diffusion map与monocle2不同，diffusion map不需要clusters等信息，不需要根据特定的属性对细胞进行分组聚类。因此，Diffusion map可以保留连续的分化轨迹。</p>
<p>Diffusion Map通过马尔科夫链的概率策略来对单细胞数据进行建模，判断细胞转移的方向，将细胞间的空间结构降维并映射为连续性结构。算法层面相对简单，比较适用于研究样本集中单个细胞的分化或者谱系起源，或者针对从不同时间点取到的相同类型的细胞进行时序分析。对于复杂的数据集或者可能存在分支的数据集，不建议使用。</p>
<p><img src="/2023/05/19/Monocle2/diffusion1.png" srcset="/img/loading.gif" lazyload><br>基本原理：</p>
<ol>
<li>输入细胞*基因的（n x G）矩阵，和monocle类似，变到高维空间</li>
<li>通过高斯核函数，高斯核函数定义成对数据点之间的相似度矩阵。从G维基因空间中找到由较高概率密度的连续路径所组成的扩散路径。</li>
<li>根据细胞对之间的转移概率，计算 n × n 的细胞间马尔可夫转移概率矩阵 。（如果对数据进行随机游走，则走到附近的数据点比走到远处的另一个数据点更有可能，也就是概率更高）</li>
<li>对马尔可夫转移概率矩阵进行特征值分解， 获得特征向量（一般取前三维度），获得细胞在不同维度的坐标。</li>
</ol>
<p>通过扩散映射将空间距离转换为状态转移的概率，对不同分化路径中的细胞进行降维和排序，确定分化细胞的随机转移方向，进而预测细胞的发育轨迹并检测稀有种群。可以与RNA velocity共同进行分析，做相互验证。</p>
<p>基于Diffusion maps算法开发的<a target="_blank" rel="noopener" href="https://doi.org/10.1093/bioinformatics/btv715">destiny软件</a>发表在bioinformatics，因为destiny包可以兼容seurat包且与其他方法相比，diffusion map算法对噪声扰动具有鲁棒性并且计算成本低(内存消耗远小于monocle)，可以将细胞间非线性的空间结构降维并映射维连续性结构。相关文献已经被引用超过1000次，也是一个稳定的分析方法了。</p>
<p>结果展示：<br><img src="/2023/05/19/Monocle2/diffusion2.png" srcset="/img/loading.gif" lazyload></p>

                
              </div>
            
            <hr/>
            <div>
              <div class="post-metas my-3">
  
    <div class="post-meta mr-3 d-flex align-items-center">
      <i class="iconfont icon-category"></i>
      

<span class="category-chains">
  
  
    
      <span class="category-chain">
        
  <a href="/categories/%E8%BD%AF%E4%BB%B6%E7%AE%97%E6%B3%95/" class="category-chain-item">软件算法</a>
  
  
    <span>></span>
    
  <a href="/categories/%E8%BD%AF%E4%BB%B6%E7%AE%97%E6%B3%95/%E7%94%9F%E4%BF%A1/" class="category-chain-item">生信</a>
  
  

  

      </span>
    
  
</span>

    </div>
  
  
    <div class="post-meta">
      <i class="iconfont icon-tags"></i>
      
        <a href="/tags/%E6%8A%80%E6%9C%AF/">#技术</a>
      
        <a href="/tags/%E4%B8%AA%E4%BA%BA%E6%88%90%E9%95%BF/">#个人成长</a>
      
    </div>
  
</div>


              
  

  <div class="license-box my-3">
    <div class="license-title">
      <div>Monocle2</div>
      <div>https://xuan-98-l.github.io/2023/05/19/Monocle2/</div>
    </div>
    <div class="license-meta">
      
        <div class="license-meta-item">
          <div>Author</div>
          <div>Xuan Liu</div>
        </div>
      
      
        <div class="license-meta-item license-meta-date">
          <div>Posted on</div>
          <div>May 19, 2023</div>
        </div>
      
      
      
        <div class="license-meta-item">
          <div>Licensed under</div>
          <div>
            
              
              
                <a target="_blank" href="https://creativecommons.org/licenses/by/4.0/">
                  <span class="hint--top hint--rounded" aria-label="BY - Attribution">
                    <i class="iconfont icon-by"></i>
                  </span>
                </a>
              
            
          </div>
        </div>
      
    </div>
    <div class="license-icon iconfont"></div>
  </div>



              
                <div class="post-prevnext my-3">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/2023/05/22/Cellchat/" title="Cellchat">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">Cellchat</span>
                        <span class="visible-mobile">Previous</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2023/05/19/Hexo-personal-design/" title="Hexo常用命令">
                        <span class="hidden-mobile">Hexo常用命令</span>
                        <span class="visible-mobile">Next</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>

    <div class="side-col d-none d-lg-block col-lg-2">
      
  <aside class="sidebar" style="margin-left: -1rem">
    <div id="toc">
  <p class="toc-header">
    <i class="iconfont icon-list"></i>
    <span>Table of Contents</span>
  </p>
  <div class="toc-body" id="toc-body"></div>
</div>



  </aside>


    </div>
  </div>
</div>





  



  



  



  



  







    

    
      <a id="scroll-top-button" aria-label="TOP" href="#" role="button">
        <i class="iconfont icon-arrowup" aria-hidden="true"></i>
      </a>
    

    
      <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">Search</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v" for="local-search-input">Keyword</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>

    

    
  </main>

  <footer>
    <div class="footer-inner">
  
    <div class="footer-content">
       <a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a> <i class="iconfont icon-love"></i> <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener"><span>Fluid</span></a> 
    </div>
  
  
  
  
</div>

  </footer>

  <!-- Scripts -->
  
  <script  src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js" ></script>
  <link  rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css" />

  <script>
    NProgress.configure({"showSpinner":false,"trickleSpeed":100})
    NProgress.start()
    window.addEventListener('load', function() {
      NProgress.done();
    })
  </script>


<script  src="https://lib.baomitu.com/jquery/3.6.0/jquery.min.js" ></script>
<script  src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js" ></script>
<script  src="/js/events.js" ></script>
<script  src="/js/plugins.js" ></script>


  <script  src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js" ></script>
  <script>
    (function (window, document) {
      var typing = Fluid.plugins.typing;
      var subtitle = document.getElementById('subtitle');
      if (!subtitle || !typing) {
        return;
      }
      var text = subtitle.getAttribute('data-typed-text');
      
        typing(text);
      
    })(window, document);
  </script>




  
    <script  src="/js/img-lazyload.js" ></script>
  




  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/tocbot/4.18.2/tocbot.min.js', function() {
    var toc = jQuery('#toc');
    if (toc.length === 0 || !window.tocbot) { return; }
    var boardCtn = jQuery('#board-ctn');
    var boardTop = boardCtn.offset().top;

    window.tocbot.init(Object.assign({
      tocSelector     : '#toc-body',
      contentSelector : '.markdown-body',
      linkClass       : 'tocbot-link',
      activeLinkClass : 'tocbot-active-link',
      listClass       : 'tocbot-list',
      isCollapsedClass: 'tocbot-is-collapsed',
      collapsibleClass: 'tocbot-is-collapsible',
      scrollSmooth    : true,
      includeTitleTags: true,
      headingsOffset  : -boardTop,
    }, CONFIG.toc));
    if (toc.find('.toc-list-item').length > 0) {
      toc.css('visibility', 'visible');
    }

    Fluid.events.registerRefreshCallback(function() {
      if ('tocbot' in window) {
        tocbot.refresh();
        var toc = jQuery('#toc');
        if (toc.length === 0 || !tocbot) {
          return;
        }
        if (toc.find('.toc-list-item').length > 0) {
          toc.css('visibility', 'visible');
        }
      }
    });
  });
</script>


  <script src=https://lib.baomitu.com/clipboard.js/2.0.11/clipboard.min.js></script>

  <script>Fluid.plugins.codeWidget();</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js', function() {
    window.anchors.options = {
      placement: CONFIG.anchorjs.placement,
      visible  : CONFIG.anchorjs.visible
    };
    if (CONFIG.anchorjs.icon) {
      window.anchors.options.icon = CONFIG.anchorjs.icon;
    }
    var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
    var res = [];
    for (var item of el) {
      res.push('.markdown-body > ' + item.trim());
    }
    if (CONFIG.anchorjs.placement === 'left') {
      window.anchors.options.class = 'anchorjs-link-left';
    }
    window.anchors.add(res.join(', '));

    Fluid.events.registerRefreshCallback(function() {
      if ('anchors' in window) {
        anchors.removeAll();
        var el = (CONFIG.anchorjs.element || 'h1,h2,h3,h4,h5,h6').split(',');
        var res = [];
        for (var item of el) {
          res.push('.markdown-body > ' + item.trim());
        }
        if (CONFIG.anchorjs.placement === 'left') {
          anchors.options.class = 'anchorjs-link-left';
        }
        anchors.add(res.join(', '));
      }
    });
  });
</script>


  
<script>
  Fluid.utils.createScript('https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js', function() {
    Fluid.plugins.fancyBox();
  });
</script>


  <script>Fluid.plugins.imageCaption();</script>

  <script  src="/js/local-search.js" ></script>





<!-- 主题的启动项，将它保持在最底部 -->
<!-- the boot of the theme, keep it at the bottom -->
<script  src="/js/boot.js" ></script>


  

  <noscript>
    <div class="noscript-warning">Blog works best with JavaScript enabled</div>
  </noscript>
</body>
</html>
